---
title: "main_and_supplement figures"
output: html_document
date: '2023-10-31'
---

# Main and Supplemental Figures for "Commensal skin bacteria exacerbate inflammation and delay healing"
#### note that this markdown doc is organised into sections: it begins with initialising the environment, loading functions, etc. and then is organised figure-by-figure through the main text and supplemental figures. All imported dataframes and objects are available for download in this repository.


# Initialise

```{r}

# Packages

library("dplyr")
library("gridExtra")
library("ggplot2")
library("ggrepel")
library("ggforce")
library("scales")
library("ggsci")
library("ggsignif")
library("ggpubr")
library("ggtext")
library("ggh4x")
library("MESS")
library("factoextra")
library("speedyseq")
library("forcats")
library("qiime2R")
library("reshape2")
library("ggpmisc")

# Global mini functions and defaults

options(scipen = 999)
'%!in%' = function(x,y)!('%in%'(x,y))
standard_error <- function(x) sd(x, na.rm=TRUE) / sqrt(length(x))
theme_set(theme_bw())

# Color schema
color_pals = c("Control1" = "#868686FF" , "Group1" = "#0073C2FF", "Group2" = "#CD534CFF", "Control2" ="#3B3B3BFF", "Group3" = "#EFC000FF")

spf_pals = c("PBS" = "#868686FF" , "Sepi44" = "#0073C2FF", "S_aureus" = "#CD534CFF", "Sepi21" = "#0B5351", "Sepi_NIH" = "darkgreen", "Sepi105" = "#7e007e", "L_reuteri" = "#EFC000FF", "B_subtilis" = "#783f8e", "E_coli" = "#4d8793", "S_xylosus" = "#8FD6DB", "Coryne" ="#f5a300", "C_accolens" = "#FF5722", "Staph_hominis930" = "#ff00ff", "Staph_hominis931" = "#8000ff", "EK_Staph_hominis930" = "#80629d", "EK_Staph_hominis931" = "#9d629d",  "Sepi44_Formalin" = "black", "Sepi_HeatKilled" = "#003B82", "Sepi_EthanolKilled" = "#007c82" , "PBS_SFB" = "black", "Sepi44_SFB" = "blue4")

abx_pals = c("Ctrl" = "#868686FF", "Abx" = "#b55757")


# Data directories

input_folder = "../data/"
current_outfolder = "../Figs"
supp_outfolder = "../Figs/supplement/"

```

# Functions for TEWL and score data

```{r}


auc_per_mouse = function(tewl_scores, day_to_exclude_past=5) {
  #' Calculates AUC for every mouse in tewl_scores type dataframe
  #' for example, a day to exclude past would be day 5 if you're trying to compare multiple cohorts at day 4
  #' by default this includes day 0 but excludes the baseline condition
    
  tewl_scores$ID = paste(tewl_scores$Organism, tewl_scores$Cohort, tewl_scores$Mouse, sep = "_")
  
  auc_df_dp = tewl_scores %>%
    filter(Day > 0 & Day < day_to_exclude_past) %>%
    group_by(Organism, ID, Cohort) %>%
    summarise(TEWL_AUC = auc(Day, TEWL, type = "spline"))
  
    auc_df_dp_score = tewl_scores %>%
    filter(Day > 1 & Day < day_to_exclude_past & !is.na(complete_score)) %>%
    group_by(Organism, ID, Cohort) %>%
    summarise(Score_AUC = auc(Day, complete_score, type = "spline"))

    aucs = left_join(auc_df_dp, auc_df_dp_score)
  
  return(aucs)
}



auc_norm = function(aucs){
  #' Normalise mouse AUCs to PBS AUC average of that cohort
  
  normalised_df = group_by(filter(aucs, Organism == "PBS"),Cohort) %>%
    summarise(PBS_TEWLmean = mean(TEWL_AUC, na.rm = TRUE), PBS_scoremean = mean(Score_AUC, na.rm = TRUE)) %>%
    left_join(aucs, ., by = c("Cohort")) %>%
    group_by(Cohort) %>%
    mutate(TEWL_AUC_norm = TEWL_AUC / PBS_TEWLmean,
           Score_AUC_norm = Score_AUC / PBS_scoremean) %>%
    select(Organism, ID, Cohort, TEWL_AUC_norm, Score_AUC_norm)
  return(normalised_df)
}



sig_labels_auc_norm = function(normalised_auc_df){
  #' Calculate p value for difference between PBS and experimental group mouse normalised AUCs
 
   df2 = normalised_auc_df
  df3 = data.frame()
  PBS_tewl = subset(df2,  Organism == "PBS")$TEWL_AUC_norm
  PBS_score = subset(df2, Organism == "PBS")$Score_AUC_norm
  
  for (organism in unique(subset(df2, Organism != "PBS")$Organism)){
    org_tewl_norm = subset(df2, Organism == organism)$TEWL_AUC_norm
    org_score_nrom = subset(df2, Organism == organism)$Score_AUC_norm
    tewl_p = wilcox.test(PBS_tewl, org_tewl_norm)$p.value
    score_p = wilcox.test(PBS_score, org_score_nrom)$p.value
    df3 = rbind(df3, data.frame("Organism" = organism, "TEWL_p" = tewl_p, "Score_p" = score_p))
  } 

df3 = df3 %>%
   mutate(TEWL_p_sci = format(TEWL_p, scientific = TRUE, digits = 3)) %>%
   mutate(Score_p_sci = format(Score_p, scientific = TRUE, digits = 3)) %>%
    mutate(TEWL_p_rounded = ifelse(TEWL_p < 0.001, "<0.001", round(TEWL_p, 3)),
         Score_p_rounded = ifelse(Score_p < 0.001, "<0.001", round(Score_p, 3)))

return(df3)
}


make_stats_db = function(tewl_db){
  #' Calculates mean and SEM for kinetic tewl and score data
  
  tewl_db <- tewl_db %>%
  group_by(Group, Day, Condition) %>%
  mutate(mean = mean(TEWL, na.rm = TRUE)) %>%
  mutate(sd = sd(TEWL, na.rm=TRUE)) %>%
  mutate(sem = standard_error(TEWL))
}


```


# 16S plotting functions

```{r}


generate_collapsed_counts_df = function(physeq_object, cutoff, taxonomic_levels_ofinterest){
  
  #' Generates the "plot_counts" df necessary for plotting bar charts where taxa are either displayed if >= cutoff or else collapsed to the taxonomic level above, all the way to phylum. 
  
  phylum_df = plot_bar(physeq_object, fill = "Phylum")$data
  phylum_df = phylum_df[phylum_df$Abundance >0,]
  
  mut_phylum_df = phylum_df
  mut_phylum_df = mut_phylum_df[mut_phylum_df$Abundance >0,] # a sanity check df to make sure we have iterated over all taxa
  mut_phylum_df$Species = sapply(mut_phylum_df$Species, function(x) {ifelse(x %in% c("uncultured bacterium","mouse gut metagenome" ,  "metagenome","uncultured rumen bacterium",  "gut metagenome", "unidentified rumen bacterium JW32", "unidentified" ), NA, x)})
  
  mut_phylum_df$Genus = sapply(mut_phylum_df$Genus, function(x) {ifelse(x %in% c("uncultured"), NA, x)})
  
  mut_phylum_df = mut_phylum_df %>%
    mutate(Species= ifelse(is.na(Genus), NA_real_, Species))
  
  plot_counts = data.frame()
  
  for (level in taxonomic_levels_ofinterest) {
    if (level == "Species"){
      for (i in unique(as.character(mut_phylum_df$Species))){
        if (is.na(i) & i %!in% c("S_uncultured_bacterium")){
          temp_counts = subset(mut_phylum_df, is.na(Species))%>% 
            group_by(Species) %>% 
            summarize(Abundance = sum(Abundance))
        } else { 
          temp_counts = subset(mut_phylum_df, Species == i)%>% 
          group_by(Species) %>% 
          summarize(Abundance = sum(Abundance))
          }
        if (temp_counts$Abundance > cutoff & !is.na(i)){#Add all instances of these species to plot_counts 
          species_plot = subset(mut_phylum_df, Species == i)[,c("Sample", "Abundance", "Species")]
          colnames(species_plot) = c( "Sample", "Abundance", "Abbr_Taxa")
          species_plot$Level = rep("Species", nrow(species_plot))
          species_plot = cbind(species_plot, subset(mut_phylum_df, Species == i)[,c("Phylum","Genus","Species")])
          plot_counts = rbind(plot_counts, species_plot ) 
          # Remove these taxa from the df so they dont get counted toward the next iteration
          mut_phylum_df_toremove = rownames(subset(mut_phylum_df, Species == i))
          mut_phylum_df = mut_phylum_df[rownames(mut_phylum_df) %!in% mut_phylum_df_toremove,]
        }
      }
    } else if (level == "Genus") {
      for (g in unique(as.character(mut_phylum_df$Genus))){
        if (is.na(g)){
          temp_genus_counts = subset(mut_phylum_df, is.na(Genus))%>% 
            group_by(Genus) %>% 
            summarize(Abundance = sum(Abundance))
        } else {
          temp_genus_counts = subset(mut_phylum_df, Genus == g)%>% 
            group_by(Genus) %>% 
            summarize(Abundance = sum(Abundance))}
        
        if (temp_genus_counts$Abundance > cutoff & !is.na(g)){
          #Add all instances of these species to plot_counts 
          genus_plot = subset(mut_phylum_df, Genus == g)[,c( "Sample", "Abundance", "Genus")]
          colnames(genus_plot) = c( "Sample", "Abundance", "Abbr_Taxa")
          genus_plot$Level = rep("Genus", nrow(genus_plot))
          genus_plot = cbind(genus_plot, subset(mut_phylum_df, Genus == g)[,c("Phylum","Genus","Species")])
          plot_counts = rbind(plot_counts, genus_plot ) 
          # Remove these taxa from the df so they dont get counted toward the next iteration
          mut_phylum_df_toremove = rownames(subset(mut_phylum_df, Genus == g))
          mut_phylum_df = mut_phylum_df[rownames(mut_phylum_df) %!in% mut_phylum_df_toremove,]
        }
      }
    } else if (level == "Phylum") {
      for (p in unique(as.character(mut_phylum_df$Phylum))){
        if (is.na(p)){
          phylum_plot = subset(mut_phylum_df, is.na(Phylum))[,c("Sample", "Abundance", "Phylum")]
          colnames(phylum_plot) = c( "Sample", "Abundance", "Abbr_Taxa")
          phylum_plot$Level = rep("Phylum", nrow(phylum_plot))
          phylum_plot = cbind(phylum_plot, subset(mut_phylum_df, is.na(Phylum))[,c("Phylum","Genus","Species")])
          plot_counts = rbind(plot_counts, phylum_plot) 
          mut_phylum_df_toremove = rownames(subset(mut_phylum_df, is.na(Phylum)))
          mut_phylum_df = mut_phylum_df[rownames(mut_phylum_df) %!in% mut_phylum_df_toremove,]
        }else {
          phylum_plot = subset(mut_phylum_df, Phylum == p)[,c("Sample", "Abundance", "Phylum")] 
          colnames(phylum_plot) = c("Sample", "Abundance", "Abbr_Taxa")
          phylum_plot$Level = rep("Phylum", nrow(phylum_plot))
          phylum_plot = phylum_plot %>%
            group_by(Sample) %>%
            mutate("Abund"  =  sum(Abundance)) %>%
            select(-Abundance) %>%
            select(Sample, Abund, Abbr_Taxa, Level) %>%
            rename(Abundance = Abund)
          phylum_plot = unique(phylum_plot)
          phylum_plot$Phylum = rep(p, nrow(phylum_plot))
          phylum_plot$Genus = rep(NA, nrow(phylum_plot))
          phylum_plot$Species = rep(NA, nrow(phylum_plot))
          plot_counts = rbind(plot_counts, phylum_plot ) 
          mut_phylum_df_toremove = rownames(subset(mut_phylum_df, Phylum == p))
          mut_phylum_df = mut_phylum_df[rownames(mut_phylum_df) %!in% mut_phylum_df_toremove,]}
      }
    }
  }
  
  return(plot_counts) 
  }


plot_collapsed_counts = function(plot_counts, sample_order = NULL){
  
  #' Plot the collapsed counts df (plot_counts) with a unique coloring scheme 
  #' To ensure consistency among figures, I have limited the taxa and colors to be used.

  top_phyla = c("Firmicutes", "Proteobacteria" ,"Actinobacteria" , "Bacteroidetes")

  phyla_colors = c("#367171", "#6a51a3",  "#7D3200", "#238b45")
  phyla_colors_pastel = c( "#DDF5F0" , "#dadaeb","#D8C7BE" , "#c7e9c0")
  

  names(phyla_colors) = top_phyla #phyla_colors is now a named list
  names(phyla_colors_pastel) = top_phyla
  
  phyla_color_ramp = list() # a list of limits to generate the color ramp between. 
  
  for (i in 1:length(names(phyla_colors))){
    tax = names(phyla_colors)[i]
    miniramp = c(as.character(phyla_colors[i]), phyla_colors_pastel[i])
    phyla_color_ramp[[tax]] <- c(as.character(phyla_colors[i]), phyla_colors_pastel[i])
  }
  
  # Giving all phyla that are not in our top 4 a light grey color, as well as the appropriate pastel color for whatever our top phyla are
  plot_counts$Color = sapply(as.character(plot_counts$Phylum), function(x) {ifelse(x %in% names(phyla_colors_pastel), phyla_colors_pastel[x], "grey20")}) 
  
  # Get the top taxa and the order of their phyla
  
  toptax =  unique(plot_counts[order(plot_counts$Abundance, decreasing = TRUE), "Abbr_Taxa"])[c(1:7)]
  toptax = toptax[toptax %!in% top_phyla] #do not include any phyla here since they get their own coloring
  
  toptax_named = list()
  for (i in toptax) {
    phy= unique(plot_counts$Phylum[plot_counts$Abbr_Taxa == i])
    toptax_named[[i]] = phy
  }
  
  phyla_order = c("Firmicutes", "Proteobacteria" ,"Actinobacteria" , "Bacteroidetes")
   
  toptax_phyla = list("Firmicutes"  = sum(toptax_named == "Firmicutes"), 
                      "Proteobacteria" = sum(toptax_named == "Proteobacteria") ,
                      "Actinobacteria" = sum(toptax_named == "Actinobacteria"),
                      "Bacteroidetes" = sum(toptax_named == "Bacteroidetes"))

  phylum_to_taxa = toptax_named
  taxa_to_phylum = setNames(names(toptax_named), toptax_named)
  toptax_ordered = c("Firmicutes",unname(taxa_to_phylum[names(taxa_to_phylum) == "Firmicutes"]),
                     "Proteobacteria" , unname(taxa_to_phylum[names(taxa_to_phylum) == "Proteobacteria"]),
                     "Actinobacteria" , unname(taxa_to_phylum[names(taxa_to_phylum) == "Actinobacteria"]),
                     "Bacteroidetes", unname(taxa_to_phylum[names(taxa_to_phylum) == "Bacteroidetes"]))

  # Remove empty phyla from toptax_phyla
  
  toptax_phyla = toptax_phyla[toptax_phyla != 0]
  
    # Generate the right color ramp 
  
  toptax_shades = list()
  for (phy in names(toptax_phyla)){
    pastel_color = unique(as.character(subset(plot_counts, Phylum == phy)$Color))
    vivid_color = as.character(phyla_colors[phy])
    if (pastel_color == "grey80") {
      print("AAHHHH THE DREADED EXCEPTION HAS OCCURRED")
    } else {
      ntax = as.numeric(toptax_phyla[phy])
      num_ramp = ntax + 1
      shades_of_phy =  colorRampPalette(c(pastel_color, vivid_color))(num_ramp)[c(2:num_ramp)]
      toptax_shades[[phy]] = shades_of_phy
    }
  }
  
  top5tax_inorder = toptax_ordered[toptax_ordered %!in% top_phyla]  #generate color palette. 
  
  taxa_to_col = list()
  
  for (tax in top5tax_inorder){
    phylum = as.character(phylum_to_taxa[tax])
    colors = as.character(toptax_shades[[phylum]])
    singlecol = colors[1]
    taxa_to_col[tax] = singlecol
    colors = colors[colors!=singlecol]
    toptax_shades[[phylum]] = colors 
  }
  
  # To highlight S. xylosus compositional differences in our data, we highlight him with a special colour

  if ("Staphylococcus xylosus" %in% names(taxa_to_col)){
    taxa_to_col[names(taxa_to_col) == "Staphylococcus xylosus"] = "#57b5b5"
  }

  # Update plot_counts to have the right color scheme
  
  for (tax in names(taxa_to_col)){
    if (tax %in% plot_counts$Abbr_Taxa){
      plot_counts <- within(plot_counts, Color[Abbr_Taxa == tax] <- as.character(taxa_to_col[tax]))
    }
  }
  
  # Change some things around in taxa order and phyla order
  
  for (ph_name in names(phyla_colors)){
    if (ph_name %!in% toptax_ordered) {toptax_ordered = c(toptax_ordered, ph_name)}
  }
  
  # Some machinsations to get everything in the right order, so that breaks and values line up. 
  
  plot_counts$Grouped_Taxa = sapply(as.character(plot_counts$Abbr_Taxa), function(x) {ifelse(x %in% toptax_ordered, x, unique(as.character(subset(plot_counts, Abbr_Taxa ==x)$Phylum)))}) #this will help us collapse the coloring later
  
  # Create levels in the right order
  alltax_without_top5 = unique(as.character(plot_counts$Grouped_Taxa))
  alltax_without_top5 = alltax_without_top5[alltax_without_top5 %!in% toptax_ordered]
  
  # Relevel and reorder 
  plot_counts$Grouped_Taxa = factor(plot_counts$Grouped_Taxa, levels = c(toptax_ordered, alltax_without_top5))
  
  plot_counts= plot_counts %>%
    arrange(factor(Grouped_Taxa, levels = c(toptax_ordered, alltax_without_top5))) #i spent 2 hrs on this
  
  manual_cols = as.character(plot_counts$Color)
  names(manual_cols) = as.character(plot_counts$Grouped_Taxa) 
  manual_cols = manual_cols[!duplicated(names(manual_cols))]
  sample_order = c(unique(as.character(plot_counts$Sample)))

  
  return(plot_counts)
  }




```


# Import QIIME2 output

```{r}

# Import metadata
# Note on naming convention here: SPF29 refers to the cohort for which there is 16S data.
# "Alex" refers to the fact that the classifier was built by Alex Poret, and is available on her github: ajporet/cutaneous_wound_microbiome. This is to distinguish it from a classifier I built using the SILVA138 database, which is not used in any of the following analyses.

spf29_metadata = read.csv(paste(input_folder,"16s_data", "spf29_metadata.csv", sep = "/"), stringsAsFactors = FALSE)

# Add classifier and build phyloseq object

spf29_alex_ps = qza_to_phyloseq(features = "../data/16s_data/DADA2_denoising_output/table.qza",
                                 taxonomy = "../data/16s_data/classifier_output/classification.qza",
                                 metadata = "../data/16s_data/spf29_metadata.tsv")


# Some finnegaling to get the sample names correct 

sample_names(spf29_alex_ps) = sample_data(spf29_alex_ps)$new_sample_id #to account for mislabelling
spf29_alex_ps@sam_data$Timepoint = factor(spf29_alex_ps@sam_data$Timepoint, levels = c("Arrival", "Prechallenge", "Endpoint"))

saveRDS(spf29_alex_ps, "../data/16s_data/classifier_output/spf29_phyloseq.rds")


# Basic decontam and filtering

filter_metrics = data.frame(sample_sums(spf29_alex_ps))
colnames(filter_metrics) = "Pre_Filtering"

spf29_alex_ps = subset_taxa(spf29_alex_ps, Kingdom != "d__Eukaryota")
spf29_alex_ps = subset_taxa(spf29_alex_ps, Kingdom != "Unassigned")
spf29_alex_ps = subset_taxa(spf29_alex_ps, !is.na(Phylum))

# Filter taxa with < 250 reads across all samples
spf29_alex_ps_filt = filter_taxa(spf29_alex_ps, function(x) sum(x) > 250, prune = TRUE)
spf29_alex_ps_filt_norm = transform_sample_counts(spf29_alex_ps_filt, function(x) x/sum(x)) #normalised

# Only skin samples
skin_samples = subset_samples(spf29_alex_ps_filt_norm, Sample_Type == "skin")
skin_samples = filter_taxa(skin_samples, function(x) sum(x) > 0, prune = TRUE)

saveRDS(skin_samples, "../data/16s_data/classifier_output/skin_samples.rds")

# Only gut samples

gut_samples = subset_samples(spf29_alex_ps_filt_norm, Sample_Type == "fecal")
gut_samples = filter_taxa(gut_samples, function(x) sum(x) > 0, prune = TRUE)

gut_samples@sam_data$Group = factor(gut_samples@sam_data$Group , levels = c("Ctrl", "Abx"))

# There is one sample in gut samples that has an experimental replicate, merge them:

gut_samples@sam_data$MergeNames = sapply(gut_samples@sam_data$SampleName, function(x) {ifelse(startsWith(x, "06_24_poop_m2"), "06_24_poop_m2", x)})
gut_samples_noreps = merge_samples(gut_samples, "MergeNames" ) # some loss of metadata but thats ok

gut_samples_noreps = filter_taxa(gut_samples_noreps, function(x) sum(x) > 0, prune = TRUE)
gut_samples_noreps_norm = transform_sample_counts(gut_samples_noreps, function(x) x/sum(x))

saveRDS(gut_samples_noreps_norm, "../data/16s_data/classifier_output/gut_samples.rds")

```


# Data Import

```{r}

# we are not including any:
# (1) Damage modulation
# (2) vendor differences
# (3) Formalin killed Sepi
# (4) Oral precolonisation with lacto
# (5) Lacto supernatant testing
# (6) cage change frequency testing

all_spf = read.csv( "../data/all_tewl_scores.csv")

all_spf$Sex = sapply(all_spf$Sex, function(x) {ifelse(startsWith(x, "F"), "F", "M")})
all_spf$Condition = sapply(all_spf$Condition, function(x) {ifelse(x %in% c("Baseline", "Recovery", "No_tapestrip"), x, "Tapestrip")})

skin_samples = readRDS("../data/16s_data/classifier_output/skin_samples.rds")
spf29_metadata = read.csv("../data/16s_data/spf29_metadata.csv", stringsAsFactors = FALSE)


```


# Figure 1 
Perturbation of the native skin microbiome by depletion or supplementation delays healing. 


## Fig 1b: Antibiotics CFU

```{r}

# CFU data

abx_cfu = read.csv("../data/Abx_Skin_CFUs_prechallenge.csv", header = TRUE)

abx_cfu$Group = factor(abx_cfu$Group, levels = c("Ctrl", "Abx"))
abx_cfu$Condition = factor(abx_cfu$Condition, levels = c("Pre-shave", "Endpoint"))

# Add pseudocounts to show plot at 0

abx_cfu$NumbertoPlot = sapply(abx_cfu$NumbertoPlot, function(x) {ifelse(x == 0, x+1, x)})

# Calculate pvalue for plotting
abx_cfu_pval = wilcox.test(subset(abx_cfu, Condition == "Endpoint" & Group == "Ctrl")$NumbertoPlot,subset(abx_cfu, Condition == "Endpoint" & Group == "Abx")$NumbertoPlot )$p.val

abx_cfu_pval  = ifelse(abx_cfu_pval < 0.001, "p<0.001", round(abx_cfu_pval, 3))

  # Sina plot
  
  ggplot(subset(abx_cfu, Condition == "Endpoint"), aes(x = Group, y = NumbertoPlot,color = Group, fill = Group))+
   stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
        labels = trans_format("log10", math_format(10^.x)))+
  scale_x_discrete(labels = c("Control", "Antibiotic"))+
  scale_fill_manual(values = abx_pals, guide = "none")+
  scale_color_manual(values = abx_pals, guide = "none")+
  labs(y = "CFU/mL/g", x ="")+
     theme_classic()+
  theme(axis.text.y = element_text(size = 12, face = "bold"),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank(), strip.text = element_blank(),
        aspect.ratio = 1.25/1)+
  annotate(geom = "text", label = abx_cfu_pval, x = 2, y = 10^6.123, parse = TRUE)
  
ggsave(paste(current_outfolder, "fig1_b.pdf", sep="/"), device = "pdf", dpi = 500, useDingbats = FALSE )



```

## Fig 1c: Scaled 16s composition plots at endpoint

```{r}

# Merge all endpoint skin samples by group 
skin_end =  subset_samples(skin_samples, Timepoint == "Endpoint")
skin_end = prune_taxa(taxa_sums(skin_end) > 0, skin_end)

merged_skin_end = merge_samples(skin_end, "Group") # just summed
merged_skin_end = transform_sample_counts(merged_skin_end, function(x) x/sum(x)) #avergaed and normalised

# Calculate scaling factors
ctrl_endpoint_scale_factor = median(subset(abx_cfu, Condition == "Endpoint" & Cohort == "29" & Group == "Ctrl")$NumbertoPlot, na.rm=T)
abx_endpoint_scale_factor = median(subset(abx_cfu, Condition == "Endpoint" & Cohort == "29" & Group == "Abx")$NumbertoPlot, na.rm=T)

# Generate collapsed counts
merged_skin_end_rel = generate_collapsed_counts_df(merged_skin_end, 0.05, c("Species", "Genus", "Phylum")) #leaves everything in
plot_counts = plot_collapsed_counts(merged_skin_end_rel)

# Prepare color palette
manual_cols = as.character(plot_counts$Color)
names(manual_cols) = as.character(plot_counts$Grouped_Taxa) 
manual_cols = manual_cols[!duplicated(names(manual_cols))]
sample_order = c(unique(as.character(plot_counts$Sample)))

# Generate scaled counts object
  
scaled_plt_counts = plot_counts %>%
  mutate(Scaled_Abundance = ifelse(Sample == "Ctrl", Abundance * ctrl_endpoint_scale_factor, Abundance * abx_endpoint_scale_factor))

# Plot   
total_comp =  ggplot(scaled_plt_counts, aes(x=factor(Sample, levels = sample_order), y= Scaled_Abundance))+
    geom_bar(aes(fill = Grouped_Taxa), stat="identity", width = 0.55)+ # makes bars
    geom_bar(stat = "identity", aes(color = Grouped_Taxa, group = Grouped_Taxa), fill = NA, lwd = 1, width = 0.55,)+ #makes outlines
       scale_fill_manual(name = element_blank() , values = manual_cols) +
  scale_x_discrete(labels = c("Control", "Antibiotic-\nTreated"))+
   scale_y_continuous(expand = c(0,0), label = comma)+
    scale_color_manual( values = manual_cols)+
    labs(x = "Sample", y = "Scaled Abundance")+
    guides(color = FALSE)+
   theme_classic()+
   theme( panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
           panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          axis.title.y = element_text(size = 12),
          panel.border = element_blank(), 
          axis.text.y = element_text(size = 10),
          axis.ticks.x = element_blank(), 
          axis.ticks.length.y = unit(0.01, "cm"),
          axis.text.x = element_text(size = 10), 
          strip.background = element_blank(),
          aspect.ratio = 1)
    
ggsave(paste(current_outfolder, "fig1_c_1.pdf", sep="/"), total_comp,  device = "pdf", dpi = 500, useDingbats = TRUE )


abx_comp = ggplot(subset(scaled_plt_counts, Sample == "Abx"), aes(x=factor(Sample, levels = sample_order), y= Scaled_Abundance))+
    geom_bar(aes(fill = Grouped_Taxa), stat="identity", width = 0.55)+ # makes bars
    geom_bar(stat = "identity", aes(color = Grouped_Taxa, group = Grouped_Taxa), fill = NA, lwd = 1, width = 0.55,)+ #makes outlines
       scale_fill_manual(name = element_blank() , values = manual_cols) +
  scale_x_discrete(labels = c("Antibiotic-\nTreated"))+
   scale_y_continuous(expand = c(0,0), label = comma)+
    scale_color_manual( values = manual_cols)+
    labs(x = "Sample", y = "Scaled Abundance")+
    guides(color = FALSE)+
   theme_classic()+
   theme( panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
           panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          axis.title.y = element_text(size = 12),
          panel.border = element_blank(), 
          axis.text.y = element_text(size = 10),
          axis.ticks.x = element_blank(), 
          axis.ticks.length.y = unit(0.01, "cm"),
          axis.text.x = element_text(size = 10), 
          strip.background = element_blank(),
          aspect.ratio = 3/1)
    
ggsave(paste(current_outfolder, "fig1_c_2.pdf", sep="/"), abx_comp,  device = "pdf", dpi = 500, useDingbats = TRUE )
    

```


## Fig 1d: Abx AUC of TEWL and score

```{r}

abx_mice = subset(all_spf, Cohort %in% c(29,34) & Organism != "Sepi44" & Group != "abx_sepi")

# Because the split is by group, not organism here, we must tweak our standard functions a little

abx_mice$ID = paste(abx_mice$Group, abx_mice$Cohort, abx_mice$Mouse, sep = "_")
  
auc_df_dp = abx_mice %>%
  filter(Condition != "Baseline" & Day < 5) %>%
  group_by(Group, ID, Cohort) %>%
  summarise(TEWL_AUC = auc(Day, TEWL, type = "spline"))

auc_df_dp_score = abx_mice %>%
  filter(Day > 1 & Day < 5 & !is.na(complete_score)) %>%
  group_by(Group, ID, Cohort) %>%
  summarise(Score_AUC = auc(Day, complete_score, type = "spline"))

abx_aucs = left_join(auc_df_dp, auc_df_dp_score)
abx_aucs = subset(abx_aucs, ID != "ctrl_pbs_29_5") #this mouse was removed from the experiment after 1 day 
  
# Normalise AUCs to control animals

normalised_abx_auc_df = group_by(filter(abx_aucs, Group == "ctrl_pbs"),Cohort) %>%
    summarise(PBS_TEWLmean = mean(TEWL_AUC, na.rm = TRUE), PBS_scoremean = mean(Score_AUC, na.rm = TRUE)) %>%
    left_join(abx_aucs, ., by = c("Cohort")) %>%
    group_by(Cohort) %>%
    mutate(TEWL_AUC_norm = TEWL_AUC / PBS_TEWLmean,
           Score_AUC_norm = Score_AUC / PBS_scoremean) %>%
    select(Group, ID, Cohort, TEWL_AUC_norm, Score_AUC_norm)

# Calculate p values for plotting

normalised_abx_auc_df$Group = factor(normalised_abx_auc_df$Group, levels = c("ctrl_pbs", "abx_pbs"))
    
df2 = normalised_abx_auc_df
abx_auc_labs = data.frame()
PBS_tewl = subset(df2,  Group == "ctrl_pbs")$TEWL_AUC_norm
PBS_score = subset(df2, Group == "ctrl_pbs")$Score_AUC_norm
  
for (group in unique(subset(df2, Group != "ctrl_pbs")$Group)){
    org_tewl_norm = subset(df2, Group == group)$TEWL_AUC_norm
    org_score_nrom = subset(df2, Group == group)$Score_AUC_norm
    tewl_p = wilcox.test(PBS_tewl, org_tewl_norm)$p.value
    score_p = wilcox.test(PBS_score, org_score_nrom)$p.value
    abx_auc_labs = rbind(abx_auc_labs, data.frame("Group" = group, "TEWL_p" = tewl_p, "Score_p" = score_p))
  } 

abx_auc_labs = abx_auc_labs %>%
   mutate(TEWL_p_sci = format(TEWL_p, scientific = TRUE, digits = 3)) %>%
   mutate(Score_p_sci = format(Score_p, scientific = TRUE, digits = 3))
  
  
  # Plot

abx_pbs_pals = c("ctr_pbs" = "#868686FF", "abx_pbs" = "#b55757")

ggplot(normalised_abx_auc_df,aes(x = Group, y = TEWL_AUC_norm, color=Group, fill=Group))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  scale_color_manual(values = abx_pbs_pals)+
  scale_fill_manual(values= abx_pbs_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1)+
  scale_x_discrete(labels = c("PBS", "Antibiotic-treated"))+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_text(data = abx_auc_labs, mapping = aes(x = Group, y = 1.3, label = round(TEWL_p, 3)), color = "black")

ggsave(paste(current_outfolder, "fig1_d_1.pdf", sep="/"),  device = "pdf", dpi = 500, useDingbats = FALSE )


ggplot(normalised_abx_auc_df,aes(x = Group, y = Score_AUC_norm, color=Group, fill=Group))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.1, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.1, seed = 123))+ # note: points were too close together for sina to handle, had to add jitter
  scale_color_manual(values = abx_pbs_pals)+
  scale_fill_manual(values= abx_pbs_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1)+
  scale_x_discrete(labels = c("PBS", "Antibiotic-treated"))+
  labs(y = "Normalized Score AUC", x = "")+
  geom_text(data = abx_auc_labs, mapping = aes(x = Group, y = 2.5, label = round(Score_p, 3)), color = "black")

ggsave(paste(current_outfolder, "fig1_d_2.pdf", sep="/"),  device = "pdf", dpi = 500, useDingbats = FALSE )


```

## Fig 1f: AUCs of mice dosed with S. xylosus


```{r}

# Calculate AUCs
xylo_aucs = auc_per_mouse(xylo)
xylo_aucs_norm = auc_norm(xylo_aucs)
xylo_aucs_labs = sig_labels_auc_norm(xylo_aucs_norm)


# Plot
ggplot(xylo_aucs_norm,aes(x = Organism, y = TEWL_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  scale_x_discrete(labels = c("PBS", expression(italic("S. xylosus"))))+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_text(data = xylo_aucs_labs, mapping = aes(x = Organism, y = 1.7, label = "<0.001"), color = "black")

ggsave(paste(current_outfolder, "fig1_f_1.pdf", sep="/"),  device = "pdf", dpi = 500, useDingbats = FALSE )

ggplot(xylo_aucs_norm,aes(x = Organism, y = Score_AUC_norm, color=Organism, fill=Organism))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =12), 
        axis.text.y = element_text(size=12), 
        axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1)+
  scale_x_discrete(labels = c("PBS", expression(italic("S. xylosus"))))+
  labs(y = "Normalized Score AUC", x = "")+
  geom_text(data = xylo_aucs_labs, mapping = aes(x = Organism, y = 3, label = "<0.001"), color = "black")

ggsave(paste(current_outfolder, "fig1_f_2.pdf", sep="/"),  device = "pdf", dpi = 500, useDingbats = FALSE )


```


## Fig 1e: S xylosus correlations with AUC

```{r}

# Read in phyloseq objects
skin_samples = readRDS("../data/16s_data/classifier_output/skin_samples.rds")
spf29_metadata = read.csv("../data/16s_data/spf29_metadata.csv", stringsAsFactors = FALSE)

# Subset out only s. xylosus
skin_xylo = subset_taxa(skin_samples, Species == "Staphylococcus xylosus")
skin_xylo_df = data.frame(t(skin_xylo@otu_table))
colnames(skin_xylo_df) = "S.xylosus_abundance"
row.names(spf29_metadata) = spf29_metadata[,1]

# Add metadata to S. xylosus relative abundance info
skin_xylo_df = merge(skin_xylo_df, spf29_metadata, by = "row.names")
skin_xylo_df = subset(skin_xylo_df, Timepoint != "Arrival")
skin_xylo_df$Timepoint = factor(skin_xylo_df$Timepoint, levels = c("Prechallenge", "Endpoint"))

# Add AUCs to metadata and S. xylosus relative abundance info
spf29_auc_norm = subset(normalised_abx_auc_df, Cohort == "29")
spf29_auc_norm$ID  = sapply(spf29_auc_norm$ID, function(x) {paste(strsplit(x, "_")[[1]][c(1,4)], collapse = "_")})
skin_xylo_end = subset(skin_xylo_df,Timepoint == "Endpoint")
skin_xylo_end$ID = sapply(skin_xylo_end$Row.names, function(x) {paste(strsplit(x, "_")[[1]][c(3,4)], collapse = "_")})
spf29_auc_norm = merge(spf29_auc_norm,skin_xylo_end , by = "ID")

# Colors
abx_pals = c("Ctrl" = "#868686FF", "Abx" = "#b55757")


# Note that this plot has been moved tgo the supplement. I leave it here for completeness
xylo_corr_tewl =  ggplot(spf29_auc_norm, aes(x = S.xylosus_abundance, y = TEWL_AUC_norm ))+
  geom_point(aes(color = Group.y), size = 6)+
  labs(y = "Normalised TEWL AUC")+
  scale_color_manual(values  = abx_pals)+
   stat_smooth( method = "lm", color = "grey40", alpha = 0.1 , lwd = 2)+
  theme_classic()+
   stat_cor(label.y = 0.8)

xylo_corr_score = ggplot(spf29_auc_norm, aes(x = S.xylosus_abundance, y = Score_AUC_norm ))+
  geom_point(aes(color = Group.y), size = 6)+
  labs(y = "Normalized Score AUC")+
  scale_color_manual(values  = abx_pals)+
   stat_smooth( method = "lm", stat = "identity", color = "grey40", alpha = 0.1, lwd = 2)+
   stat_cor(label.y = 0.8)+
  theme_classic()

ggsave(paste(supp_outfolder, "fig1_e_1.pdf", sep="/"), xylo_corr_tewl,  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )
ggsave(paste(current_outfolder, "fig1_e_2.pdf", sep="/"), xylo_corr_score,  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```


# Figure 2
Human skin commensals including S. epidermidis delay barrier repair. 

## Figure 2a-b: Panel of all tested bacteria

```{r}

# Subset the correct data

 all_bugs = subset(all_spf, Cohort %!in% c(2, 4, 8, 10, 12, 13, 15, 16, 20, 21, 23, 24, 29, 31,32,33) & Organism %!in% c("Sepi44_Formalin", "L_reuteri_supernatant","Sepi_EthanolKilled", "EK_Staph_hominis930", "EK_Staph_hominis931", "Sepi_HeatKilled", "Coryne", "C_tuberculostericum", "Staph_hominis931", "Sepi_NIH", "Sepi105", "Sepi44_1x") & Group %!in% c("5X_PBS", "10x_Sepi1e07") & Day < 5)


# AUC normalised to each group's own PBS

all_bugs_aucs = auc_per_mouse(all_bugs)
all_bugs_aucs_norm = auc_norm(all_bugs_aucs)
aucs_labs = sig_labels_auc_norm(all_bugs_aucs_norm)

all_bugs_aucs_norm$Organism = factor(all_bugs_aucs_norm$Organism, levels= c( "PBS", "Sepi44", "S_xylosus", "S_aureus", "Staph_hominis930", "C_accolens", "L_reuteri", "B_subtilis", "E_coli"))

aucs_labs$Organism = factor(aucs_labs$Organism, levels= c("Sepi44", "S_aureus", "S_xylosus", "Staph_hominis930", "C_accolens", "L_reuteri", "B_subtilis", "E_coli"))

# Prepare data for plotting and calculate pvalues

fig1b = subset(all_bugs_aucs_norm, Organism != "PBS")
fig1b_labs = aucs_labs
fig1b_labs$TEWL_p_rounded = round(fig1b_labs$TEWL_p,3)
fig1b_labs$TEWL_p_rounded = sapply(fig1b_labs$TEWL_p_rounded, function(x) {ifelse(x <0.001, "p<0.001", x)})
fig1b_labs$Score_p_rounded = round(fig1b_labs$Score_p,3)
fig1b_labs$Score_p_rounded = sapply(fig1b_labs$Score_p_rounded, function(x) {ifelse(x <0.001, "p<0.001", x)})

# Set color defaults
fg1b_cols = c( rep("navajowhite4",length(unique(fig1b$Organism))))
names(fg1b_cols) = c("Sepi44", "S_xylosus", "S_aureus", "Staph_hominis930", "C_accolens","L_reuteri", "B_subtilis", "E_coli" )
fg1b_cols[names(fg1b_cols) == "Sepi44"] = "#0073C2FF"
fg1b_cols[names(fg1b_cols) == "S_xylosus"] = "#57b5b5"

# Plot TEWL and Score AUCs with cropped axes

tewl_lim = 2
score_lim  = 3
offset = 0.1

fig1b$TEWL_AUC_norm2 <- sapply(fig1b$TEWL_AUC_norm, function(x) {ifelse(abs(x) > tewl_lim, tewl_lim + offset, x)})
fig1b$Score_AUC_norm2 <- sapply(fig1b$Score_AUC_norm, function(x) {ifelse(abs(x) > score_lim, score_lim + offset, x)})

fig1b$tewl_flag <- abs(fig1b$TEWL_AUC_norm) > tewl_lim # add a flag variable to help filter for points outside the limits
fig1b$score_flag <- abs(fig1b$Score_AUC_norm) > score_lim

fig1b = fig1b %>%
  group_by(Organism) %>%
  mutate(tewl_norm_median = median(TEWL_AUC_norm, na.rm = T),
         score_norm_median = median(Score_AUC_norm, na.rm = T))

# Plot 
  ggplot()+
  geom_hline( yintercept = 1, color = "grey60", size = 0.5)+
   geom_sina(data = subset(fig1b, !tewl_flag), aes( x = Organism, y = TEWL_AUC_norm, fill = Organism, color = Organism), maxwidth = .7, scale = "area", seed = 1,size = 4, alpha = .5, method = "d", position = position_jitter(width = 0.2, seed = 123)) + 
  geom_sina(data = subset(fig1b, !tewl_flag),aes( x = Organism, y = TEWL_AUC_norm, fill = Organism, color = Organism), maxwidth = .7, scale = "area", seed = 1, size = 4, shape = 1, color = "black", stroke = .4, method = "d", position = position_jitter(width = 0.2, seed = 123))+
    stat_summary(data = fig1b, fun.y = median, geom = "errorbar", aes(x = Organism, y = TEWL_AUC_norm, ymax = ..y.., ymin = ..y..), color = "black", width = .8,size = 1, alpha = 0.9)+
  geom_dotplot(data = subset(fig1b, tewl_flag), aes( x = Organism,y = TEWL_AUC_norm2, fill = Organism),  binaxis = "y", stackdir = "centerwhole", color = "black", dotsize = 2.5, alpha  = 0.5, binwidth = 0.04) +
  theme_light()+
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.border = element_blank(),
          axis.text.x = element_text(size =12, angle = 45, vjust = 0.8, hjust = 1), 
          axis.text.y = element_text(size=12),
          axis.title = element_text(size = 20), 
          legend.position = "none")+
   annotation_custom(grid::linesGrob(x = c(0, 0), gp = grid::gpar(lwd = 3))) +
  scale_color_manual(values = fg1b_cols)+
  scale_fill_manual(values= fg1b_cols)+
  scale_y_continuous(breaks = c(0.5, 1.0, 1.5, 2.0))+
  scale_x_discrete(labels = c(expression(italic("S. epidermidis")), expression(italic("S. xylosus")), expression(italic("S. aureus")), expression(italic("S. hominis")), expression(italic("C. accolens")), expression(italic("L. reuteri")), expression(italic("B. subtilis")), expression(italic("E. coli"))))+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_text(data = fig1b_labs, mapping = aes(x = Organism, y = 2.3, label = TEWL_p_rounded))

  
  ggsave(paste(current_outfolder, "fig2_a.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )
 
  ggplot()+
  geom_hline( yintercept = 1, color = "grey60", size = 0.5)+
    geom_hline( yintercept = 1, color = "grey60", size = 0.5)+
   geom_sina(data = subset(fig1b, !score_flag), aes( x = Organism, y = Score_AUC_norm, fill = Organism, color = Organism), maxwidth = .7, scale = "area", seed = 1,size = 4, alpha = .5, method = "d", position = position_jitter(width = 0.2, seed = 123)) + 
  geom_sina(data = subset(fig1b, !score_flag),aes( x = Organism, y = Score_AUC_norm, fill = Organism, color = Organism), maxwidth = .7, scale = "area", seed = 1, size = 4, shape = 1, color = "black", stroke = .4, method = "d", position = position_jitter(width = 0.2, seed = 123))+
    stat_summary(data = fig1b, fun.y = median, geom = "errorbar", aes(x = Organism, y = Score_AUC_norm, ymax = ..y.., ymin = ..y..), color = "black", width = .8,size = 1, alpha = 0.9)+
  geom_dotplot(data = subset(fig1b, score_flag), aes( x = Organism,y = Score_AUC_norm2, fill = Organism),  binaxis = "y", stackdir = "center", color = "black", dotsize = 3, alpha  = 0.5, binwidth = 0.04) +
    theme_light()+
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.border = element_blank(),
          axis.text.x = element_text(size =12, angle = 45, vjust = 0.9, hjust = 1), 
          axis.text.y = element_text(size=12),
          axis.title = element_text(size = 20), 
          legend.position = "none")+
  annotation_custom(grid::linesGrob(x = c(0, 0), gp = grid::gpar(lwd = 3))) +
  scale_color_manual(values = fg1b_cols)+
  scale_fill_manual(values= fg1b_cols)+
  scale_x_discrete(labels = c(expression(italic("S. epidermidis")), expression(italic("S. xylosus")), expression(italic("S. aureus")), expression(italic("S. hominis")), expression(italic("C. accolens")), expression(italic("L. reuteri")), expression(italic("B. subtilis")), expression(italic("E. coli"))))+
  labs(y = "Normalized Score AUC", x = "")+
    scale_y_continuous(limits = c(0.3, 3.8), breaks = seq(0.5,3,0.5))+
  geom_text(data = fig1b_labs, mapping = aes(x = Organism, y = 3.5, label = Score_p_rounded))

ggsave(paste(current_outfolder, "fig2_b.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

## Fig 2c-d: S epi strains and killed S. epi comparisons

```{r}

# Subset and merge all Sepi cohorts and killed cohorts

two_sepi_df = subset(all_spf, Cohort %in% c(12, 14,27, 36, 17, 18, 19, 21, 23, 26, 37) & Organism %in% c("PBS", "Sepi_NIH", "Sepi105", "Sepi_HeatKilled", "Sepi_EthanolKilled") & Group %!in% c("control0", "group0", "7x_PBS-F", "5X_PBS", "5X_Sepi1e09", "10x_Sepi1e07", "Shominis930", "Sepi_SFB",  "PBS_SFB", "abx_sepi" ,"abx_pbs", "pbs_nots",  "sepi_nots"))

# Calculate AUCs, normalise, and calculate p values
sepi_aucs = auc_per_mouse(subset(two_sepi_df, Day <5))
sepi_acus_norm = auc_norm(sepi_aucs)
sepi_aucs_labs = sig_labels_auc_norm(sepi_acus_norm)
sepi_acus_norm = subset(sepi_acus_norm, Organism %!in%  c("PBS", "Sepi44"))
sepi_aucs_labs = subset(sepi_aucs_labs, Organism %!in%  c( "Sepi44"))


sepi_acus_norm$Organism = factor(sepi_acus_norm$Organism, levels = c(  "Sepi105", "Sepi_NIH", "Sepi_EthanolKilled", "Sepi_HeatKilled" ))
sepi_aucs_labs$Organism = factor(sepi_aucs_labs$Organism, levels = c( "Sepi105", "Sepi_NIH", "Sepi_EthanolKilled", "Sepi_HeatKilled" ))


# Once again, we will plot this data on cropped axes, but this time it is only the score AUCs that need cropping

score_lim  = 3
offset = 0.3

sepi_acus_norm$Score_AUC_norm2 <- sapply(sepi_acus_norm$Score_AUC_norm, function(x) {ifelse(abs(x) > score_lim, score_lim + offset, x)})
sepi_acus_norm$score_flag <- abs(sepi_acus_norm$Score_AUC_norm) > score_lim

  
ggplot(sepi_acus_norm, aes(x = Organism, y = TEWL_AUC_norm, color = Organism, fill = Organism))+
    geom_hline( yintercept = 1, color = "grey60", size = 0.5)+
   geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 4, alpha = .5, method = "d", position = position_jitter(width = 0.2, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 4, shape = 1, color = "black", stroke = .4, method = "d", position = position_jitter(width = 0.2, seed = 123))+
    stat_summary( fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..), color = "black", width = .8,size = 1, alpha = 0.9)+
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          axis.text.x = element_text(size =12, hjust = 1, vjust = 1, angle = 45), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none")+ #aspect.ratio = 1/1.5
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  scale_y_continuous(limits = c(0.5, 2), breaks = seq(0.5,2.0,0.5))+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_text(data = sepi_aucs_labs, mapping = aes(x = Organism, y = 1.8, label = TEWL_p_rounded), color = "black")

ggsave(paste(current_outfolder, "fig2_c.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

  
ggplot(sepi_acus_norm, aes(x = Organism, y = Score_AUC_norm, color = Organism, fill = Organism))+
      geom_hline( yintercept = 1, color = "grey60", size = 0.5)+
     geom_sina(data = subset(sepi_acus_norm, !score_flag), aes( x = Organism, y = Score_AUC_norm, fill = Organism), maxwidth = .7, scale = "area", seed = 1,size = 4, alpha = .5, method = "d", position = position_jitter(width = 0.2, seed = 123)) + 
  geom_sina(data = subset(sepi_acus_norm, !score_flag), aes( x = Organism, y = Score_AUC_norm, fill = Organism), maxwidth = .7, scale = "area", seed = 1, size = 4, shape = 1, color = "black", stroke = .4, method = "d", position = position_jitter(width = 0.2, seed = 123))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..), width = .8,size = 1, alpha = 0.9, color = "black")+
  geom_dotplot(data = subset(sepi_acus_norm, score_flag), aes( x = Organism,y = Score_AUC_norm2, fill = Organism),  binaxis = "y", stackdir = "centerwhole", color = "black", dotsize = 3, alpha  = 0.5, binwidth = 0.04) +
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          axis.text.x = element_text(size =12,  hjust = 1, vjust = 1, angle = 45), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none")+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  scale_x_discrete()+
  labs(y = "Normalized Score AUC", x = "")+
  geom_text(data = sepi_aucs_labs, mapping = aes(x = Organism, y = 3.5, label = Score_p_rounded), color = "black")

ggsave(paste(current_outfolder, "fig2_d.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

## Figure 3
Application of S. epidermidis following barrier damage induces an innate inflammatory immune response.


#Fig 3b-c:  Immune cells

```{r}

# Import data and clean up names
spf35_gate2 = read.csv("../data/gs3_freqs.csv", header = TRUE)
spf35_d4_melt = melt(spf35_gate2, id = c("Group", "Mouse", "Day"))
colnames(spf35_d4_melt)[c(4,5)] = c("Population", "Freq")
spf35_d4_melt = subset(spf35_d4_melt,  Population != "X" & !endsWith(Freq, ".fcs") & Population %!in% c("Sac_TEWL", "Sac_Score", "Live_CD45_of_Total", "Live_CD45"))

spf35_d4_melt$Mouse = factor(spf35_d4_melt$Mouse)
spf35_d4_melt$Freq = as.numeric(spf35_d4_melt$Freq)

# Calculate mean and standard dev for each population
spf35_d4_melt_stat = spf35_d4_melt %>%
  group_by(Group, Day, Population) %>%
  mutate(mean = mean(Freq),
    sd = standard_error(Freq))

# Calculate p balues 

spf35_d4_melt_stat_labs = spf35_d4_melt %>%
  group_by (Day, Population, Group) %>%
  summarise(values = list(Freq)) %>%
  group_by(Day, Population) %>% 
  summarise( p_value= wilcox.test(values[[1]],values[[2]])$p.value) %>%
  mutate(sig_label = case_when(p_value > 0.05 ~ "n.s.",
                                p_value > 0.01 & p_value < 0.05 ~ "*",
                                p_value > 0.001 & p_value < 0.01 ~ "**",
                                p_value > 0 & p_value < 0.001 ~ "***"))


spf35_d4_melt_stat$Population = factor(spf35_d4_melt_stat$Population, levels = c("Neutrophils", "Monocytes", "Macrophages", "DC"))

immne_cell_pops = ggplot(subset(spf35_d4_melt_stat, Day == 4 & Population != "lc_freq"), aes(x = Population, y = Freq ))+
  geom_point(aes(color = Group, fill = Group), size = 4, position = position_jitterdodge(seed =123), alpha = 0.6)+
  geom_point(aes(color = Group, fill = Group), size = 4, position = position_jitterdodge(seed = 123),shape = 1, color = "black")+
  geom_boxplot(aes(color=Group, fill = Group), outlier.colour = NA, alpha = 0.2, color = "grey40", lwd = .3)+
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          axis.text.x = element_text(size =12, hjust = 0.5, vjust = 0.3), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none",
          aspect.ratio = 1/1.5)+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values = spf_pals)+
  scale_x_discrete(labels = c("Neutrophils", "Monocytes", "Macrophages", "DCs"))+
  labs(x = "", y = "% of CD45+")+
  geom_text(data = subset(spf35_d4_melt_stat_labs, Day ==4 & Population != "lc_freq"), mapping = aes(x = Population, y = 60, label = round(p_value, 3)))

ggsave(paste(current_outfolder, "fig3_b.pdf", sep="/"), immne_cell_pops,  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

# Correlation of neutrophils with severity score on day 4

spf35_gate2_counts = read.csv("../data/gs3_counts.csv", header = TRUE)

neut_corr =  ggplot(subset(spf35_gate2_counts, Day == 4), aes(x = Sac_Score, y = Neutrophils))+
    stat_smooth( method = "lm", stat = "identity", color = "grey40", alpha = 0.1, lwd = 2)+
  geom_point(aes(color = Group), size = 6, alpha = 0.6)+
  scale_color_manual(values = spf_pals)+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
        labels = trans_format("log10", math_format(10^.x)))+
  labs(x = "Disease score", y = "Neutrophil Count")+
  theme_classic()+
   stat_cor(label.y = 4.5)

ggsave(paste(current_outfolder, "fig3_c.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

## Fig 3d:  IL-17 data

```{r}

# Import IL17 data
sepi_17 = read.csv("../data/il_17_sepi.csv", header = TRUE)
sepi_17 = subset(sepi_17, Group %in% c("PBS", "Sepi44"))

# Calculate p values
il17_pval = wilcox.test(subset(sepi_17, Group == "PBS")$IL_17_Protein,subset(sepi_17, Group == "Sepi44")$IL_17_Protein )$p.val
il17_pval = ifelse(il17_pval < 0.001, "p<0.001", round(il17_pval,3))

ggplot(sepi_17, aes(x = Group, y = IL_17_Protein, fill = Group, color = Group))+
     stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.12, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.12, seed = 123))+
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          axis.text.x = element_text(size =12, hjust = 0.3, vjust = 0.3), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none",
          aspect.ratio = 1.25/1)+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  scale_x_discrete(labels = c("PBS", expression(italic("S. epidermidis 44"))))+
  labs(y = "IL-17A (relative to toal protein)", x = "")+
    annotate(geom = "text", label = il17_pval, x = 2, y = 4.9, parse = TRUE)

ggsave(paste(current_outfolder, "fig3_d.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

## Fig 3f: pathway enrichment bubble plot

```{r}

# Imoprt data

pathways = read.csv("/Users/Veda/Dropbox (MIT)/Lieberman Lab/Personal lab notebooks/V/Commensal_Mouse/paper_data/data/immune_pathways_forVK.csv", header = TRUE)

immune_bubbles =   ggplot(pathways, aes(x = Cluster, y = reorder(Description, foldchange, sum)))+
  geom_point(aes(color = foldchange, size = Count))+
  scale_size_continuous(range = c(2,10))+
  scale_color_gradient(low = "#A5D7E8", high = "#576CBC")+
  labs(x = "", y = "")+
  theme_classic()+
    theme(aspect.ratio = 1.5/1)

ggsave(paste(current_outfolder, "fig3_f.pdf", sep="/"), immune_bubbles,  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```


## Fig. 3e: stacked bars of genes

```{r}

# Import data
immune_bars = read.csv("../data/immune_total_genes_up_down.csv", header = TRUE)
immune_bars$time = factor(immune_bars$time)

ggplot(immune_bars, aes(x = time))+
  geom_bar(aes(y = tot_up), fill = "grey40", stat = "identity", alpha =0.8)+
  geom_bar(aes(y = immune_up), fill = "#663695", stat = "identity", alpha = 0.8)+
  geom_text( aes(x = time, y = tot_up, label = tot_up), position = position_dodge(width = 1),cvjust = -0.5, size = 6)+
  geom_text( aes(x = time, y = immune_up-1, label = immune_up), position = position_dodge(width = 1),vjust = 0.9, size = 6, color = "white")+
  theme_classic()+
  coord_flip()+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
        labels = trans_format("log10", math_format(10^.x)))+
  theme(aspect.ratio = 1.5/1)

ggsave(paste(current_outfolder, "fig3_e_1.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

ggplot(immune_bars, aes(x = time))+
  geom_bar(aes(y = tot_down), fill = "grey40", stat = "identity", alpha =0.8)+
  geom_bar(aes(y = immune_down), fill = "#663695", stat = "identity", alpha = 0.8)+
  geom_text( aes(x = time, y = tot_down, label = tot_down), position = position_dodge(width = 1),cvjust = -0.5, size = 6)+
  geom_text( aes(x = time, y = immune_down, label = immune_down), position = position_dodge(width = 1), size = 6, color = "white")+
  theme_classic()+
  coord_flip()+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
        labels = trans_format("log10", math_format(10^.x)))+
  theme(aspect.ratio = 1.5/1)

ggsave(paste(current_outfolder, "fig3_e_2.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

```

# Figure 4

S. epidermidis application after barrier damage increases and prolongs expression of epithelial cell proliferation pathways

## Fig. 4a: Pathway enrichment bubbles

```{r}


prolif_path = read.csv("../data/lm_curated_pathways_plot.csv", header = TRUE)

prolif_path$Description = factor(prolif_path$Description, levels = rev(c("glycosaminoglycan binding","endopeptidase activity","extracellular matrix binding","threonine-type endopeptidase activity" ,"structural constituent of cytoskeleton","actin filament binding","extracellular matrix structural constituent")))

prolif_bubbles =  ggplot(prolif_path, aes(x = Cluster, y = Description))+
  geom_point(aes(color = foldchange, size = Count))+
  scale_size_continuous(range = c(2,10))+
  scale_color_gradient(low = "#9DC08B", high = "#40513B")+
  labs(x = "", y = "")+
  theme_classic()+
    theme(legend.position = "bottom")

ggsave(paste(current_outfolder, "fig4_a.pdf", sep="/"), prolif_bubbles,  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )



```

# Supplement!

## Fig 1
S. epidermidis induced inflammation and delayed barrier repair at multiple timepoints

### Fig 1a-b: Healing in PBS control mice

```{r}


# Subset data
allpbs = subset(all_spf, Group %!in% c("control0", "group0", "5X_PBS", "5X_Sepi1e09", "10x_Sepi1e07",  "7x_PBS-F", "Lacto_fed_PBS" , "Cage_change_daily_PBS",  "Sepi_PBS", "abx_pbs") & Organism == "PBS" & Cohort %!in% c(4,8,10,13,31,32))

mean_baseline = mean(subset(allpbs, Condition == "Baseline")$TEWL, na.rm=T)

ggplot(subset(allpbs, Day > 1), aes(x = factor(Day), y = TEWL, fill = Organism))+
  geom_jitter(aes(color = "Organism"), size = 2, width = 0.25, alpha =0.4)+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
                   width = .3,size = 1.25, color = "black")+
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          axis.text.x = element_text(size =12, hjust = 1), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none",
          aspect.ratio = 0.8)+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  geom_hline(yintercept = mean_baseline, color = "grey40")+
  labs(y = "TEWL", x = "")

ggsave(paste(supp_outfolder, "supp_fig1a.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


ggplot(subset(allpbs, Day > 1), aes(x = factor(Day), y = complete_score, fill = Organism))+
  geom_jitter(aes(color = Organism), size = 2, width = 0.25, alpha =0.4, height = 0.25)+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
                   width = .2,size = 1.25, color = "black")+
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          axis.text.x = element_text(size =12, hjust = 1), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none",
          aspect.ratio = 0.8)+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
    geom_hline(yintercept = 0, color = "grey40")+
  labs(y = "Score", x = "")


ggsave(paste(supp_outfolder, "supp_fig1b.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

### Fig 1d: Epidermal thickness vs score


```{r}


thicc = read.csv("../data/supplement/forvk_ave_thickness_histology.csv")

ggplot(subset(thicc, group %in% c("PBS", "Sepi44")), aes(x = thickness, y = score))+
    stat_smooth( method = "lm", stat = "identity", color = "grey40", alpha = 0.1, lwd = 2)+
  geom_jitter(aes(color = group), size = 6, alpha = 0.6)+
  scale_color_manual(values = spf_pals)+
  labs(x = "Epidermal Thickness", y = "Severity Score")+
  scale_y_continuous(limits = c(0,8), breaks = seq(0, 8,1))+
  theme_classic()+
   stat_cor(label.y = 4.5)

ggsave(paste(supp_outfolder, "supp_fig1d.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

```


### Fig 1e: Histology scores for sepi and pbs mice


```{r}

histopath = read.csv("../data/supplement/pathscores.csv", header = TRUE)

histopath$complete_score = histopath$Crust.Score + histopath$Dermal.Neutrophil.Score + histopath$Subcutaneous.Fat.Score
histopath_usable = subset(histopath, Timepoint %in% c("Endpoint", "24h") & Condition == "tapestrip")

# Calculate p values
histo_pval = wilcox.test(subset(histopath, Cohort != 25 & Group == "PBS")$complete_score, subset(histopath, Cohort != 25 & Group == "Sepi44")$complete_score)$p.val

ggplot(histopath_usable, aes(x = Group, y = complete_score, color = Group, Fill = Group))+
   stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.25, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.25, seed = 123))+
  facet_wrap(.~Timepoint)+
  scale_color_manual(values = spf_pals)+
  theme_classic()+
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_blank())+
  labs (x = "", y = "Pathology Score")+
  geom_signif(comparisons = list(c("PBS", "Sepi44")), tip_length = 0, color = "black")

ggsave(paste(supp_outfolder, "supp_fig1e.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

## Figure 2:

Oral antibiotic treatment depleted the skin microbiome with minimal impact on gut microbiome

### Fig 2a: Day by day antibiotic treated mice

```{r}

# TEWL and Score

abx_mice = subset(all_spf, Cohort %in% c(29,34) & Organism != "Sepi44" & Group != "abx_sepi")

abx_mice_stats = abx_mice %>%
  group_by(Group, Day, Condition) %>%
  mutate(mean = mean(TEWL, na.rm = TRUE)) %>%
  mutate(sd = sd(TEWL, na.rm=TRUE)) %>%
  mutate(sem = standard_error(TEWL)) %>%
  mutate(score_mean = mean(complete_score, na.rm = TRUE)) %>%
  mutate(score_sem = standard_error(complete_score)) %>%
  mutate(Day = replace(Day, Condition == "Tapestrip", 0.5),
         Group = case_when(Group == "abx_pbs" ~ "Abx",
                           Group == "ctrl_pbs" ~ "Ctrl")) # wow that worked lmao


abx_mice_stats$Group = factor(abx_mice_stats$Group, levels = c("Ctrl", "Abx"))

abx_tewl = ggplot(abx_mice_stats)+
  geom_point(aes(x = Day, y = mean, group=Group,color = Group), size = 4)+
  scale_color_manual(values = abx_pals, labels = c(expression("PBS"), expression("Abx + PBS")))+
  scale_fill_manual(values = abx_pals, guide = "none")+
  geom_line(aes(x = Day, y = mean, color = Group), alpha = 0.5, size = 1 )+
  geom_ribbon(aes(x = Day, y = TEWL, ymin = mean - sem, ymax = mean+sem, color = Group, fill = Group), alpha = 0.1, color = NA)+
  scale_x_continuous(breaks = seq(0,4,1), expand = c(0,0))+
  scale_y_continuous(breaks = seq(0,90,10))+
  labs(x = "Day", y = "TEWL")+
    theme_classic() +
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =16), 
        axis.text.y = element_text(size=16),
        axis.title = element_text(size = 20), 
        axis.ticks.x = element_blank())+
  theme(legend.position = c(0.2,0.1), 
        legend.direction = "horizontal",
       legend.title = element_blank() )


ggsave(paste(supp_outfolder, "supp_fig2_a.pdf", sep=""), abx_tewl,  device = "pdf", dpi = 500, useDingbats = FALSE )

abx_score = ggplot(subset(abx_mice_stats, Day > 1))+
  geom_point(aes(x = Day, y = score_mean, group=Group,color = Group), size = 4)+
  scale_color_manual(values = abx_pals, labels = c(expression("PBS"), expression("Abx + PBS")))+
  scale_fill_manual(values = abx_pals, guide = "none")+
  geom_line(aes(x = Day, y = score_mean, color = Group), alpha = 0.5, size = 1 )+
  geom_ribbon(aes(x = Day, y = TEWL, ymin = score_mean - score_sem, ymax = score_mean+score_sem, color = Group, fill = Group), alpha = 0.1, color = NA)+
  scale_x_continuous(breaks = seq(0,4,1), expand = c(0,0))+
  scale_y_continuous(breaks = seq(0,9,1), limits = c(0,9))+
  labs(x = "Day", y = "Severity Score")+
    theme_classic() +
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =16), 
        axis.text.y = element_text(size=16),
        axis.title = element_text(size = 20), 
        axis.ticks.x = element_blank())+
  theme(legend.position = c(0.2,0.1), 
        legend.direction = "horizontal",
       legend.title = element_blank() )


ggsave(paste(supp_outfolder, "supp_fig2_b.pdf", sep="/"), abx_score,  device = "pdf", dpi = 500, useDingbats = FALSE )



```


### Fig. 2 c : Skin composition of individual mice

```{r}

# Subset the correct samples

skin_end =  subset_samples(skin_samples, Timepoint == "Endpoint")
skin_end = prune_taxa(taxa_sums(skin_end) > 0, skin_end)
skin_end_rel_collapse = generate_collapsed_counts_df(skin_end, 0.05,c("Species", "Genus", "Phylum") )

# Generate collapsed counts object
skin_end_plot = plot_collapsed_counts(skin_end_rel_collapse)
skin_end_plot$Group = sapply(skin_end_plot$Sample, function(x) {ifelse(strsplit(x, "_")[[1]][3] == "abx", "Abx", "Ctrl")})
skin_end_plot$Group  = factor(skin_end_plot$Group , levels = c("Ctrl", "Abx"))

# Color scheme
manual_cols = as.character(skin_end_plot$Color)
names(manual_cols) = as.character(skin_end_plot$Grouped_Taxa) 
manual_cols = manual_cols[!duplicated(names(skin_end_plot))]

# Plot  
ggplot(skin_end_plot, aes(x=Sample, y= Abundance))+
    geom_bar(aes(fill = Grouped_Taxa), stat="identity")+
    geom_bar(stat = "identity", aes(color = Grouped_Taxa, group = Grouped_Taxa), fill = NA, lwd = 1)+
       scale_fill_manual(name = element_blank() , values = manual_cols) +
    scale_color_manual( values = manual_cols)+
    scale_y_continuous( expand = c(0,0)) +
  facet_wrap(Group~., scales = "free_x")+
    labs(x = "Sample", y = "Relative Abundance")+
    guides(color = FALSE)+
   theme_classic()+
   theme( panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
           panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          axis.title.y = element_text(size = 12),
          panel.border = element_blank(), axis.text.y = element_text(size = 8),
          axis.ticks.x = element_blank(), axis.ticks.length.y = unit(0.01, "cm"),
          axis.text.x = element_blank(), strip.background = element_blank())  
  
    ggsave(paste(supp_outfolder, "supp_fig2c.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )



```


### Fig 2f: Alpha diversity on the skin

```{r}

# alpha diversity analysis must be conducted on unclean data that includes singletons =

spf29_phylo = readRDS("../data/16s_data/classifier_output/spf29_phyloseq.rds")
spf29_skin = subset_samples(spf29_phylo, Sample_Type == "skin" & Timepoint != "Arrival")
spf29_skin <- prune_taxa(taxa_sums(spf29_skin) > 0, spf29_skin)

# Calculate richness
skin_rich =  plot_richness(spf29_skin, measures = c("Simpson"), color = "Group")+
  facet_wrap(.~Timepoint, scales = "free_x")

skin_rich = data.frame(skin_rich$data)
skin_rich$Group = factor(skin_rich$Group, levels = c("Ctrl", "Abx"))

time.labs = c("Pre-shave", "Endpoint")
names(time.labs) = c("Prechallenge", "Endpoint")

abx_pals = c("Ctrl" = "#868686FF", "Abx" = "#b55757")

# Calculate p values 
skin_rich_pval = wilcox.test(subset(skin_rich, Timepoint == "Endpoint" & Group == "Ctrl")$value,subset(skin_rich, Timepoint == "Endpoint" & Group == "Abx")$value )$p.val
skin_rich_pval  = ifelse(skin_rich_pval < 0.001, "p<0.001", round(skin_rich_pval, 3))

# Plot
ggplot(subset(skin_rich, Timepoint == "Endpoint"), aes(x = Group, y = value, fill = Group, color = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  scale_x_discrete(labels = c("Control", "Antibiotic-treated"))+
  theme_classic()+
  scale_color_manual(values = abx_pals)+
  scale_fill_manual(values = abx_pals)+
  labs(x = "", y = "Simpson's Index of Diversity (1-D)")+
 annotate(geom = "text", label = skin_rich_pval, x = 2, y = 1.1, parse = TRUE)+
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        aspect.ratio = 1.25/1, 
        legend.position = "none",
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 16))+
  scale_y_continuous(limits = c(0,1.25), breaks = seq(0,1,0.25))

ggsave(paste(supp_outfolder, "supp_fig2_f.pdf", sep="/"),  device = "pdf", dpi = 500 ,useDingbats = FALSE )


```


### Fig 2: Gut Composition of indiivudal mice


```{r}

# Read in phyloseq object

gut_samples_noreps_norm = readRDS("../data/16s_data/classifier_output/gut_samples.rds")

# Only Endpoint Samples
gut_endpoint = subset_samples(gut_samples_noreps_norm, Timepoint ==3)

# Generate collapsed counts df
gut_rel_collapse = generate_collapsed_counts_df(gut_endpoint, 0.05,c("Species", "Genus", "Phylum") )
gut_plot = plot_collapsed_counts(gut_rel_collapse)
gut_plot$Group = sapply(gut_plot$Sample, function(x) {ifelse(strsplit(x, "_")[[1]][4] %in% c("m1", "m2", "m3", "m4","m5"), "Abx", "Ctrl")})
gut_plot$Group  = factor(gut_plot$Group , levels = c("Ctrl", "Abx"))

# Color palette
manual_cols = as.character(gut_plot$Color)
names(manual_cols) = as.character(gut_plot$Grouped_Taxa) 
manual_cols = manual_cols[!duplicated(names(gut_plot))]

# Plot
ggplot(gut_plot, aes(x=Sample, y= Abundance))+
    geom_bar(aes(fill = Grouped_Taxa), stat="identity")+
    geom_bar(stat = "identity", aes(color = Grouped_Taxa, group = Grouped_Taxa), fill = NA, lwd = 1)+
       scale_fill_manual(name = element_blank() , values = manual_cols) +
    scale_color_manual( values = manual_cols)+
    scale_y_continuous( expand = c(0,0)) +
  facet_wrap(Group~., scales = "free_x")+
    labs(x = "Sample", y = "Relative Abundance")+
    guides(color = FALSE)+
   theme_classic()+
   theme( panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
           panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          axis.title.y = element_text(size = 12),
          panel.border = element_blank(), axis.text.y = element_text(size = 8),
          axis.ticks.x = element_blank(), axis.ticks.length.y = unit(0.01, "cm"),
          axis.text.x = element_blank(), strip.background = element_blank())  
  
    ggsave(paste(supp_outfolder, "supp_fig2g.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```



### Fig 2: Gut alpha diversity

```{r}

# alpha diversity analysis must be conducted on unclean data that includes singletons 

# Read in phyloseq object and subset only endpoint samples
# Note: this *is* the untrimmed data, so ignore warnings about untrimmed data
spf29_phylo = readRDS("../data/16s_data/classifier_output/spf29_phyloseq.rds")
spf29_gut = subset_samples(spf29_phylo, Sample_Type == "fecal" & Timepoint == "Endpoint")

spf29_gut@sam_data$MergeNames = sapply(spf29_gut@sam_data$SampleName, function(x) {ifelse(startsWith(x, "06_24_poop_m2"), "06_24_poop_m2", x)})
spf29_gut_merge = merge_samples(spf29_gut, "MergeNames") # some loss of metadata but thats ok
spf29_gut_merge <- prune_taxa(taxa_sums(spf29_gut_merge) > 0, spf29_gut_merge)

# Calcuylate simpson diversity 
gut_rich =  plot_richness(spf29_gut_merge, measures = c("Simpson"), color = "Group")+
  facet_wrap(.~Timepoint, scales = "free_x")

gut_rich = data.frame(gut_rich$data)
gut_rich$Group = sapply(gut_rich$samples, function(x) {ifelse(strsplit(x, "_")[[1]][4] %in% c("m1", "m2", "m3", "m4","m5"), "Abx", "Ctrl")})

gut_rich$Group = factor(gut_rich$Group , levels = c("Ctrl", "Abx"))
gut_rich_pval = wilcox.test(subset(gut_rich, Group == "Ctrl")$value,subset(gut_rich, Group == "Abx")$value )$p.val
gut_rich_pval  = ifelse(gut_rich_pval < 0.001, "p<0.001", round(gut_rich_pval, 3))

# Plot

ggplot(gut_rich, aes(x = Group, y = value, color= Group, fill = Group))+
stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  scale_x_discrete(labels = c("Control", "Antibiotic-treated"))+
  theme_classic()+
  scale_color_manual(values = abx_pals)+
  scale_fill_manual(values = abx_pals)+
  labs(x = "", y = "Simpson's Index of Diversity (1-D)")+
 annotate(geom = "text", label = gut_rich_pval, x = 2, y = 1.1, parse = TRUE)+
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        aspect.ratio = 1.25/1, 
        legend.position = "none",
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 16))+
  scale_y_continuous(limits = c(0,1.25), breaks = seq(0,1,0.25))


  ggsave(paste(supp_outfolder, "supp_fig2h.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```


### Fig 2: qPCR differences in microbial load in gut and skin

```{r}


qpcr = read.csv("../data/supplement/skin_gut_16s_au.csv", header = TRUE)

qpcr = subset(qpcr, Source %in% c("Skin", "Fecal"))
qpcr$Group = factor(qpcr$Group, levels = c("Ctrl", "Abx"))

ggplot(subset(qpcr, Timepoint == "Endpoint" & Source == "Fecal"), aes(x = Group, y = AU, fill = Group, color = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  scale_x_discrete(labels = c("Control", "Antibiotic"))+
  theme_classic()+
  scale_color_manual(values = abx_pals)+
  scale_fill_manual(values = abx_pals)+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
        labels = trans_format("log10", math_format(10^.x)))+
  labs(x = "", y = "Au")+
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        aspect.ratio = 1.25/1, 
        legend.position = "none",
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 16))+
  geom_signif(comparisons = list(c("Ctrl", "Abx")), step = 0, tip_length = 0, color = "black")
  

ggsave(paste(supp_outfolder, "supp_fig2i.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


ggplot(subset(qpcr, Timepoint == "Endpoint" & Source == "Skin"), aes(x = Group, y = AU, fill = Group, color = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  scale_x_discrete(labels = c("Control", "Antibiotic"))+
  theme_classic()+
  scale_color_manual(values = abx_pals)+
  scale_fill_manual(values = abx_pals)+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
        labels = trans_format("log10", math_format(10^.x)))+
  labs(x = "", y = "Au")+
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        aspect.ratio = 1.25/1, 
        legend.position = "none",
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 16))+
  geom_signif(comparisons = list(c("Ctrl", "Abx")), step = 0, tip_length = 0, color = "black")
  

ggsave(paste(supp_outfolder, "supp_fig2e.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

  

```

## Figure 3:

Mice exposed to S. aureus after tape-stripping do not heal 4 days post-damage


## Fig 3 a-b: TEWL and score kinetics as well as AUC values

```{r}

# Subset cohorts 

mrsa = subset(all_spf, Cohort %in% c("26", "36") & Organism %in% c("PBS", "S_aureus"))

mrsa_stats = mrsa %>%
  group_by(Organism, Day, Condition) %>%
  mutate(mean = mean(TEWL, na.rm = TRUE)) %>%
  mutate(sd = sd(TEWL, na.rm=TRUE)) %>%
  mutate(sem = standard_error(TEWL)) %>%
  mutate(score_mean = mean(complete_score, na.rm = TRUE)) %>%
  mutate(score_sem = standard_error(complete_score))

mrsa$Day = factor(mrsa$Day)

ggplot(subset(mrsa, Condition != "Baseline"& Day !=0), aes(x = Day, y = TEWL, color = Organism, fill = Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..,),
                 width = .6,size = 1, alpha = 0.9, position = position_dodge(width = 1))+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_dodge(width = 1)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_dodge(width = 1))+
    scale_color_manual(values = spf_pals)+
    scale_fill_manual(values = spf_pals)+
    theme_classic()

 
ggsave(paste(supp_outfolder, "supp_fig3_a_1.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

 
 ggplot(subset(mrsa, Day %!in% c(0,1)), aes(x = Day, y = complete_score, color = Organism, fill = Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..,),
                 width = .6,size = 1, alpha = 0.9, position = position_dodge(width = 1))+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_dodge(width = 1)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_dodge(width = 1))+
    scale_color_manual(values = spf_pals)+
    scale_fill_manual(values = spf_pals)+
    theme_classic()


ggsave(paste(supp_outfolder, "supp_fig3_a_2.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

 # Calculate AUCs and p values
mrsa$Day = as.numeric(as.character(mrsa$Day))
mrsa_aucs = auc_per_mouse(mrsa)
mrsa_aucs_norm = auc_norm(mrsa_aucs)
mrsa_aucs_labs = sig_labels_auc_norm(mrsa_aucs_norm)

ggplot(mrsa_aucs_norm,aes(x = Organism, y = TEWL_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  scale_x_discrete(labels = c("PBS", expression(italic("S. aureus"))))+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_text(data = mrsa_aucs_labs, mapping = aes(x = Organism, y = 1.5, label = TEWL_p_rounded), color = "black")

ggsave(paste(supp_outfolder, "supp_fig3_b_1.pdf", sep="/"),  device = "pdf", dpi = 500, useDingbats = FALSE )

ggplot(mrsa_aucs_norm,aes(x = Organism, y = Score_AUC_norm, color=Organism, fill=Organism))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =12), 
        axis.text.y = element_text(size=12), 
        axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1)+
  scale_x_discrete(labels = c("PBS", expression(italic("S. aureus"))))+
  labs(y = "Normalized Score AUC", x = "")+
  geom_text(data = xylo_aucs_labs, mapping = aes(x = 2, y = 3, label = "<0.001"), color = "black")

ggsave(paste(supp_outfolder, "supp_fig3_b_2.pdf", sep="/"),  device = "pdf", dpi = 500, useDingbats = FALSE )


```

## Figure 4:

Application of S. epidermidis to intact skin had no effect on skin barrier function or subsequent response to tape-stripping damage

### Fig 4b : Application of S. epi to depliated, undamaged skin

```{r}

# Subset cohorts
sepinots = subset(all_spf, Cohort %in% c(1,25) & Day < 5 & Group %!in% c("control1", "group1"))
sepinots$Condition = rep("No_tapestrip", nrow(sepinots))

# Calculate mean, sd, SEM of TEWL and score

sepinots_stats = sepinots %>%
  group_by(Organism, Day) %>%
  mutate(mean = mean(TEWL, na.rm = TRUE)) %>%
  mutate(sd = sd(TEWL, na.rm=TRUE)) %>%
  mutate(sem = standard_error(TEWL)) %>%
  mutate(score_mean = mean(complete_score, na.rm = TRUE)) %>%
  mutate(score_sem = standard_error(complete_score)) %>%
  mutate(Day = replace(Day, Condition == "Tapestrip", 0.5)) # wow that worked lmao

# Plot
ggplot(subset(sepinots_stats, Day >1))+
  geom_point(aes(x = Day, y = mean, group=Organism,color = Organism), size = 4)+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values = spf_pals, guide = "none")+
  geom_line(aes(x = Day, y = mean, color = Organism), alpha = 0.5, size =1 )+
  geom_ribbon(aes(x = Day, y = TEWL, ymin = mean - sem, ymax = mean+sem, color = Organism, fill = Organism), alpha = 0.1, color = NA)+
  scale_x_continuous(breaks = seq(0,4,1), expand = c(0,0))+
  scale_y_continuous(breaks = seq(0,16,2))+
  labs(x = "Day", y = "TEWL")+
    theme_classic() +
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =16), 
        axis.text.y = element_text(size=16),
        axis.title = element_text(size = 20), 
        axis.ticks.x = element_blank())+
  theme(legend.position = c(0.2,0.1), 
        legend.direction = "horizontal",
       legend.title = element_blank() )

ggsave(paste(supp_outfolder, "supp_fig4_b_1.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


 ggplot(sepinots_stats)+
  geom_point(aes(x = Day, y = score_mean, group=Organism,color = Organism), size = 4)+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values = spf_pals, guide = "none")+
  geom_line(aes(x = Day, y = score_mean, color = Organism), alpha = 0.5 , size = 1)+
  geom_ribbon(aes(x = Day, y = TEWL, ymin = score_mean - score_sem, ymax = score_mean+score_sem, color = Organism, fill = Organism), alpha = 0.1, color = NA)+
  scale_x_continuous(breaks = seq(2,4,1), expand = c(0,0), limits = c(2,4))+
  scale_y_continuous(breaks = seq(0,9,1), limits = c(0,9))+
  labs(x = "Day", y = "Severity Score")+
  theme_classic() +
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =16), 
        axis.text.y = element_text(size=16),
        axis.title = element_text(size = 20), 
        axis.ticks.x = element_blank())

ggsave(paste(supp_outfolder, "supp_fig4_b_2.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

### Fig. 4d: Precolonisation of healthy skin with S. epi does not improve healing outcomes to barrier damage

```{r}

# Subset cohorts
topical_precol = subset(all_spf, Cohort %in% c(24,33))
topical_precol = subset(topical_precol, Group %!in% c("lacto_lactosepi44", "lacto_lacto"))

# Color palette
precol_pals = c("PBS_PBS" = "#868686FF" , "PBS_Sepi" = "#0073C2FF",  "Sepi_Sepi" = "#44677e", "Sepi_PBS" = "#004d80")


# Have to tweak our standard AUC and normalisation functions to group mice differently

topical_precol$ID = paste(topical_precol$Group, topical_precol$Mouse, sep = "_")
  
auc_df_dp = topical_precol %>%
    filter(Condition != "Baseline" & Day < 5) %>%
    group_by(Group, ID, Cohort) %>%
    summarise(TEWL_AUC = auc(Day, TEWL, type = "spline"))
  
auc_df_dp_score = topical_precol %>%
    filter(Day > 1 & Day < 5 & !is.na(complete_score)) %>%
    group_by(Group, ID, Cohort) %>%
    summarise(Score_AUC = auc(Day, complete_score, type = "spline"))

precol_aucs = left_join(auc_df_dp, auc_df_dp_score)

# Normalise to PBS/PBS mice
precol_aucs_norm = group_by(filter(precol_aucs, Group == "PBS_PBS"),Cohort) %>%
    summarise(PBS_TEWLmean = mean(TEWL_AUC, na.rm = TRUE), PBS_scoremean = mean(Score_AUC, na.rm = TRUE)) %>%
    left_join(precol_aucs, ., by = c("Cohort")) %>%
    group_by(Cohort) %>%
    mutate(TEWL_AUC_norm = TEWL_AUC / PBS_TEWLmean,
           Score_AUC_norm = Score_AUC / PBS_scoremean) %>%
    select(Group, ID, Cohort, TEWL_AUC_norm, Score_AUC_norm)
    
df2 = precol_aucs_norm
df3 = data.frame()
PBS_tewl = subset(df2,  Group == "PBS_PBS")$TEWL_AUC_norm
PBS_score = subset(df2, Group == "PBS_PBS")$Score_AUC_norm
  
for (Group in unique(subset(df2, Group != "PBS_PBS")$Group)){
  org_tewl_norm = subset(df2, Group == Group)$TEWL_AUC_norm
  org_score_nrom = subset(df2, Group == Group)$Score_AUC_norm
  tewl_p = wilcox.test(PBS_tewl, org_tewl_norm)$p.value
  score_p = wilcox.test(PBS_score, org_score_nrom)$p.value
  df3 = rbind(df3, data.frame("Group" = Group, "TEWL_p" = tewl_p, "Score_p" = score_p))
}
  
  precol_aucs_norm$Group = factor(precol_aucs_norm$Group, levels = c("PBS_PBS", "Sepi_PBS", "PBS_Sepi", "Sepi_Sepi" ))
  
# Plot
ggplot(precol_aucs_norm, aes(x = Group, y = TEWL_AUC_norm, color = Group, fill = Group))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.25, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.25, seed = 123))+
    scale_color_manual(values = precol_pals)+
    scale_fill_manual(values = precol_pals)+
    theme_classic()+
    geom_signif(comparisons = list(c("PBS_PBS", "Sepi_PBS"),c("PBS_Sepi", "Sepi_Sepi"), c("PBS_PBS", "PBS_Sepi"), c("PBS_PBS", "Sepi_Sepi") ), step =0.1, color = "black", tip_length = 0 )
  
ggsave(paste(supp_outfolder, "supp_fig4_d_1.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

  
ggplot(precol_aucs_norm, aes(x = Group, y = Score_AUC_norm, color = Group, fill = Group))+
       stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.25, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.25, seed = 123))+
    scale_color_manual(values = precol_pals)+
    scale_fill_manual(values = precol_pals)+
      theme_classic()+
    geom_signif(comparisons = list(c("PBS_PBS", "Sepi_PBS"),c("PBS_Sepi", "Sepi_Sepi"), c("PBS_PBS", "PBS_Sepi"), c("PBS_PBS", "Sepi_Sepi") ), step =0.1, color = "black", tip_length = 0 )


ggsave(paste(supp_outfolder, "supp_fig4_d_2.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

## Figure 5:
S. epidermidis applied after damage induces T cell response

### Fig 5a-b: Full S. epi timecourse over 7 days

```{r}

# Basic S. epi data from all relevant cohorts

sepi_cohorts = unique(subset(all_spf, Organism == "Sepi44" & Cohort %!in% c(2, 4, 8, 10, 13, 15, 16, 20, 24,26,32,33))$Cohort)

sepi_df = subset(all_spf, Cohort %in% sepi_cohorts & Organism %in% c("PBS", "Sepi44") & Group %!in% c("control0", "group0", "7x_PBS-F", "5X_PBS", "5X_Sepi1e09", "10x_Sepi1e07", "Shominis930", "Sepi_SFB",  "PBS_SFB", "abx_sepi" ,"abx_pbs", "sepi_nots", "pbs_nots" ))

# Kinetics of Sepi

# Collapse mouse #s across groups and calculate mean, SEM
sepi_stats = sepi_df %>%
  group_by(Organism, Day, Condition) %>%
  mutate(mean = mean(TEWL, na.rm = TRUE)) %>%
  mutate(sd = sd(TEWL, na.rm=TRUE)) %>%
  mutate(sem = standard_error(TEWL)) %>%
  mutate(score_mean = mean(complete_score, na.rm = TRUE)) %>%
  mutate(score_sem = standard_error(complete_score))

#Plot
ggplot(subset(sepi_stats, Condition != "Baseline"))+
  geom_point(aes(x = Day, y = mean, group=Organism,color = Organism), size = 4)+
  scale_color_manual(values = spf_pals, labels = c(expression("PBS"), expression(italic("S. epidermidis"))))+
  scale_fill_manual(values = spf_pals, guide = "none")+
  geom_line(aes(x = Day, y = mean, color = Organism), alpha = 0.5, size = 1 )+
  geom_ribbon(aes(x = Day, y = TEWL, ymin = mean - sem, ymax = mean+sem, color = Organism, fill = Organism), alpha = 0.1, color = NA)+
  scale_x_continuous(breaks = seq(0,7,1), expand = c(0,0))+
  scale_y_continuous(breaks = seq(0,90,10))+
  labs(x = "Day", y = "TEWL")+
    theme_classic() +
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =16), 
        axis.text.y = element_text(size=16),
        axis.title = element_text(size = 20), 
        axis.ticks.x = element_blank())+
  theme(legend.position = c(0.2,0.1), 
        legend.direction = "horizontal",
       legend.title = element_blank() )

ggsave(paste(supp_outfolder, "supp_fig5a.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


ggplot(subset(sepi_stats, Condition != "Baseline" & Day > 1))+
  geom_point(aes(x = Day, y = score_mean, group=Organism,color = Organism), size = 4)+
  scale_color_manual(values = spf_pals, labels = c(expression("PBS"), expression(italic("S. epidermidis"))))+
  scale_fill_manual(values = spf_pals, guide = "none")+
  geom_line(aes(x = Day, y = score_mean, color = Organism), alpha = 0.5, size = 1 )+
  geom_ribbon(aes(x = Day, y = TEWL, ymin = score_mean - score_sem, ymax = score_mean+score_sem, color = Organism, fill = Organism), alpha = 0.1, color = NA)+
  scale_x_continuous(breaks = seq(2,7,1), expand = c(0,0), limits = c(2,7))+
  scale_y_continuous(breaks = seq(0,9,1), limits = c(0,9))+
  labs(x = "Day", y = "Severity score")+
    theme_classic() +
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =16), 
        axis.text.y = element_text(size=16),
        axis.title = element_text(size = 20), 
        axis.ticks.x = element_blank())+
  theme(legend.position = c(0.2,0.1), 
        legend.direction = "horizontal",
       legend.title = element_blank() )
  
 ggsave(paste(supp_outfolder, "supp_fig5b.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

### Fig 5c: T-cells data

```{r}

spf1_par = read.csv("../data/supplement/spf1_parent_freqs.csv", header = T)

# Color palette
color_pals = c("Control1" = "#868686FF" , "Group1" = "#0073C2FF","Group0" = "#0073C2FF", "Group2" = "#CD534CFF", "Control0" ="#3B3B3BFF", "Group3" = "#EFC000FF")


params = colnames(spf1_par)
params = params[params %!in% c("Tube", "Mouse", "Sex", "Tissue", "Group", "GroupSex", "Stripper")]


# Get SEM and mean for all populations
spf1_par_stats <- spf1_par %>%
  group_by(Group, Tissue) %>%
  mutate(across(all_of(params), 
                mean, 
                .names="{.col}_mean"))%>%
  mutate(across(params,
                standard_error,
                .names="{.col}_sem")) %>%
  mutate(CD4_CD8_ratio = CD4_of_CD45CD3/CD8_of_CD45CD3)

# Calculate p values for bulk T cells
tcell_pval = wilcox.test(subset(spf1_par_stats, Group %in% c("Control1") & Tissue == "Skin")$CD45_CD3_of_Live , subset(spf1_par_stats, Group %in% c("Group1") & Tissue == "Skin")$CD45_CD3_of_Live)$p.val

#Plot
ggplot(subset(spf1_par_stats, Group %in% c("Control1", "Group1") & Tissue == "Skin"), aes(x = Group, y = CD45_CD3_of_Live, color = Group, fill = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  labs(y= "% T cells")+
  scale_color_manual(values = color_pals)+
  annotate(geom = "text", x = 2, y = 2.5, label = round(tcell_pval,3), parse = T)+
  scale_x_discrete(labels = c("PBS", "Sepi44"))+
  theme_classic()+
  theme(strip.background = element_rect(fill = NA, color = NA), legend.position = "none", aspect.ratio = 1)+
  theme(panel.grid.major = element_blank(), axis.text.x = element_text(size =16, angle = 45, hjust = 1), axis.text.y = element_text(size=16), axis.title = element_text(size = 12), legend.text.align = 0)


ggsave(paste(supp_outfolder, "supp_fig5c_1.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

 
# Plot the lack of T cells in non-tapestripped controls - this is for figure supp. 4b
 
ggplot(subset(spf1_par_stats, Group %in% c("Control0", "Group0") & Tissue == "Skin"), aes(x = Group, y = CD45_CD3_of_Live, color = Group, fill = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  labs(y= "% T cells")+
  scale_color_manual(values = color_pals)+
   geom_signif(comparisons = list(c("Control0", "Group0")))+
  scale_x_discrete(labels = c("PBS", "Sepi44"))+
  theme_classic()+
  theme(strip.background = element_rect(fill = NA, color = NA), legend.position = "none", aspect.ratio = 1)
 
ggsave(paste(supp_outfolder, "supp_fig4_b_3.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )

 # Plot:
 
 # 1. CD4 and CD8 cells
 
 tcellsubs = subset(spf1_par, Tissue == "Skin" & Group %in% c("Control1", "Group1") )[,c(4,6,7, 10, 11)]
 tcell_subs_melt = melt(tcellsubs, id = c("Mouse", "Group"))
 colnames(tcell_subs_melt)[c(3,4)] = c("Population", "PercTcell")
 
 tcellsubs_labs = tcell_subs_melt %>%
  group_by (Group, Population) %>%
  summarise(group_vals = list(PercTcell)) %>%
  group_by(Population) %>% 
  summarise( p_value= wilcox.test(group_vals[[1]],group_vals[[2]])$p.value) %>%
  mutate(sig_label = case_when(p_value > 0.05 ~ "n.s.",
                                p_value > 0.01 & p_value < 0.05 ~ "*",
                                p_value > 0.001 & p_value < 0.01 ~ "**",
                                p_value > 0 & p_value < 0.001 ~ "***"),
         pval_round = round(p_value,3))
 
ggplot(tcell_subs_melt, aes(x = Population, y = PercTcell, color = Group, fill = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9, position = position_dodge(width = 1))+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitterdodge(jitter.width = 0.25, dodge.width = 1, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitterdodge(jitter.width = 0.25, dodge.width = 1, seed = 123))+
  scale_fill_manual(values = color_pals)+
  scale_color_manual(values = color_pals)+
geom_signif(comparisons = list(c("Control1", "Group1"), step = 0.1))+
  theme_classic()+
  theme(strip.background = element_rect(fill = NA, color = NA), legend.position = "none", aspect.ratio = 1/2)+
  theme(panel.grid.major = element_blank(), axis.text.x = element_text(size =16, angle = 45, hjust = 1), axis.text.y = element_text(size=16), axis.title = element_text(size = 12), legend.text.align = 0)+
    geom_text(data = tcellsubs_labs, mapping = aes(x = Population, y = 70, label = pval_round), size = 3)

 
 ggsave(paste(supp_outfolder, "supp_fig5_c_2.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )
 
 # 2. GD T cell
 
 gdtcell =  subset(spf1_par, Tissue == "Skin" & Group %in% c("Control1", "Group1") )[,c(8, 10, 11)]
 
ggplot(gdtcell, aes(x = Group, y = TCRg_of_DN, color = Group, fill = Group))+
  stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9, position = position_dodge(width = 1))+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitterdodge(jitter.width = 0.25, dodge.width = 1, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitterdodge(jitter.width = 0.25, dodge.width = 1, seed = 123))+
  scale_fill_manual(values = color_pals)+
  scale_color_manual(values = color_pals)+
geom_signif(comparisons = list(c("Control1", "Group1"), step = 0.1), tip_length = 0)+
  theme_classic()+
  theme(strip.background = element_rect(fill = NA, color = NA), legend.position = "none", aspect.ratio = 2/1)+
  theme(panel.grid.major = element_blank(), axis.text.x = element_text(size =16, angle = 45, hjust = 1), axis.text.y = element_text(size=16), axis.title = element_text(size = 12), legend.text.align = 0)

 
  ggsave(paste(supp_outfolder, "supp_fig5_c_3.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )
  
  

```


## Figure 6:

S. epidermidis increased IL-17A protein in skin


### Fig 6a: Multiplex ELISA

```{r}

multi_elisa = read.csv("../data/supplement/multiplex_summary_toplot.csv")[,-1]
multi_elisa_melt = melt(multi_elisa, id = c("tube", "cohort", "group", "matrix"))
colnames(multi_elisa_melt)[c(3,5,6)] = c("Group","Protein", "value")

# Calculate pvals for all proteins
multi_elisa_labs = multi_elisa_melt %>%
  group_by (Group, Protein) %>%
  summarise(group_vals = list(value)) %>%
  group_by(Protein) %>% 
  summarise( p_value= wilcox.test(group_vals[[1]],group_vals[[2]])$p.value) %>%
  mutate(sig_label = case_when(p_value > 0.05 ~ "n.s.",
                                p_value > 0.01 & p_value < 0.05 ~ "*",
                                p_value > 0.001 & p_value < 0.01 ~ "**",
                                p_value > 0 & p_value < 0.001 ~ "***"))

# gotta bonferroni the elisa w FDR 0.05


multi_bh = multi_elisa_labs[order(multi_elisa_labs$p_value),]
multi_bh$k_m_a = (seq(1,nrow(multi_bh),1)/nrow(multi_bh))* 0.05 #this is the critical value
multi_bh$BH = multi_bh$p_value < multi_bh$k_m_a

# Split up plot into high and low values to improve plotting visisbility
multi_elisa_melt = na.omit(multi_elisa_melt)
multi_elisa_melt = multi_elisa_melt %>%
  group_by(Protein) %>%
  mutate(useless = mean(value),
    cutoff = case_when(useless > 20 ~ "HI",
                            useless < 20 ~ "LO"))

multi_elisa_melt$Protein = as.character(multi_elisa_melt$Protein)

ggplot(multi_elisa_melt, aes(x = Protein, y = value ))+
  geom_boxplot(aes(color=Group, fill = Group), alpha = 0.5, color = "grey20", lwd = .3)+
  theme_classic()+
    theme(panel.grid.major = element_blank(),
          strip.background = element_blank(),
          axis.text.x = element_text(size =12, hjust = 1, vjust = 0.5, angle = 45), 
          axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
          legend.position = "none",
          aspect.ratio = 1/1.5)+
  facet_wrap(cutoff~., scales = "free")+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values = spf_pals)+
  labs(x = "", y = "Protein ng/mL norm to total")

ggsave(paste(supp_outfolder, "supp_fig6a.pdf", sep="/"),  device = "pdf", dpi = 500,height = 6, width = 8 ,useDingbats = FALSE )


```

##Fig7 : Immune volcanoes {generated in python}
## Fig 8: Flow Gating {generated in FlowJo}

## Response to reviewers


## Fig X : Sepi applied once

```{r}


sepi1x = subset(all_spf, Cohort %in% c(36,37) & Organism %in% c("Sepi44_1x", "PBS"))

sepi1x_stats = sepi1x %>%
  group_by(Organism, Day, Condition) %>%
  mutate(mean = mean(TEWL, na.rm = TRUE)) %>%
  mutate(sd = sd(TEWL, na.rm=TRUE)) %>%
  mutate(sem = standard_error(TEWL)) %>%
  mutate(score_mean = mean(complete_score, na.rm = TRUE)) %>%
  mutate(score_sem = standard_error(complete_score))
  # mutate(Day = replace(Day, Condition == "Tapestrip", 0.5)) # wow that worked lmao


sepi1x_auc = auc_per_mouse(sepi1x)
sepi1x_auc_norm = auc_norm(sepi1x_auc)

x_pals = c("PBS" = "#868686FF" , "Sepi44_1x" = "#0073C2FF")

ggplot(sepi1x_auc_norm,aes(x = Organism, y = TEWL_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = x_pals)+
  scale_fill_manual(values= x_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_signif(comparisons = list(c("PBS", "Sepi44_1x")))


ggplot(sepi1x_auc_norm,aes(x = Organism, y = Score_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = x_pals)+
  scale_fill_manual(values= x_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized Score AUC", x = "")+
  geom_signif(comparisons = list(c("PBS", "Sepi44_1x")))

```

## S. epi dose response curve

```{r}

sepi_cohorts = unique(subset(all_spf, Organism == "Sepi44" & Cohort %!in% c(4, 8, 10, 13, 15, 16, 20, 24,25, 28,31,32,33,34,36,37))$Cohort)

sepi_dose_response = subset(all_spf, Cohort %in% sepi_cohorts & Organism %in% c("PBS", "Sepi44") & Group %!in% c("control0", "group0", "7x_PBS-F", "5X_PBS", "5X_Sepi1e09", "Shominis930", "Sepi_SFB",  "PBS_SFB", "abx_sepi" ,"abx_pbs" ))

# Have to manually do AUC calculations to include dose data

 sepi_dose_response$ID = paste(sepi_dose_response$Organism, sepi_dose_response$Cohort, sepi_dose_response$Mouse, sep = "_")
  
sepi_dose_auc_tewl = sepi_dose_response %>%
  filter(Condition != "Baseline" & Day < 5) %>%
  group_by(Organism, ID, Cohort, Dose) %>%
  summarise(TEWL_AUC = auc(Day, TEWL, type = "spline"))
  
sepi_dose_auc_score = sepi_dose_response %>%
filter(Day > 1 & Day < 5 & !is.na(complete_score)) %>%
group_by(Organism, ID, Cohort, Dose) %>%
summarise(Score_AUC = auc(Day, complete_score, type = "spline"))

sepi_dose_auc = left_join(sepi_dose_auc_tewl, sepi_dose_auc_score)
  

# Now normalise by cohort and dose
  

  normalised_dose_auc = group_by(filter(sepi_dose_auc, Organism == "PBS"),Cohort) %>%
    summarise(PBS_TEWLmean = mean(TEWL_AUC, na.rm = TRUE), PBS_scoremean = mean(Score_AUC, na.rm = TRUE)) %>%
    left_join(sepi_dose_auc, ., by = c("Cohort")) %>%
    group_by(Cohort) %>%
    mutate(TEWL_AUC_norm = TEWL_AUC / PBS_TEWLmean,
           Score_AUC_norm = Score_AUC / PBS_scoremean) %>%
    select(Organism, ID, Cohort, Dose, TEWL_AUC_norm, Score_AUC_norm)


  #Plot only s epi, no PBS necessary
  
  plotdf = subset(normalised_dose_auc, Organism == "Sepi44")
  plotdf$Dose = factor(plotdf$Dose, levels = c(10000000, 100000000, 1000000000))
  
ggplot(plotdf,aes(x = Dose, y = TEWL_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) +
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_x_discrete(labels= c("1e7", "1e8", "1e9"))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized TEWL AUC", x = "")


ggplot(plotdf,aes(x = Dose, y = Score_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
   scale_x_discrete(labels= c("1e7", "1e8", "1e9"))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized Score AUC", x = "")



```

## 1e08 comparison of s epi, s aureus and lacto

```{r}

 low_dose = subset(all_spf, Cohort %in% c(2,3,9) & Organism %!in% c("Sepi21", "Coryne", "S_xylosus") & Dose != "1000000000")

low_dose_auc = auc_per_mouse_d1(low_dose)
low_dose_auc_norm = auc_norm(low_dose_auc)


low_dose_auc_norm$Organism = factor(low_dose_auc_norm$Organism, levels = c("PBS", "L_reuteri", "Sepi44", "S_aureus"))

plotdf = subset(low_dose_auc_norm, Organism != "PBS")

ggplot(plotdf,aes(x = Organism, y = TEWL_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  geom_hline(yintercept = 1)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18, angle = 45, hjust = 1), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_signif(comparisons = list(c("L_reuteri", "Sepi44")), color = "black", step = 0.1)


ggplot(plotdf,aes(x = Organism, y = Score_AUC_norm, color=Organism, fill=Organism))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
   geom_hline(yintercept = 1)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18, angle = 45, hjust = 1), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized Score AUC", x = "")+
  geom_signif(comparisons = list(c("L_reuteri", "Sepi44")), color = "black", step = 0.1)

# Because we only did 1 rep of sepi and saureus, can we combine them?

plotdf$Genus = sapply(plotdf$Organism, function(x) {ifelse(x == "L_reuteri", "Lacto", "Staph")})

ggplot(plotdf,aes(x = Genus, y = TEWL_AUC_norm, color=Organism, fill=Organism))+
    # stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 # width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
  geom_hline(yintercept = 1)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18, angle = 45, hjust = 1), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized TEWL AUC", x = "")+
  geom_signif(comparisons = list(c("Lacto", "Staph")), color = "black", step = 0.1)


ggplot(plotdf,aes(x = Genus, y = Score_AUC_norm, color=Organism, fill=Organism))+
    # stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
    #              width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.3, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.3, seed = 123))+
  scale_color_manual(values = spf_pals)+
  scale_fill_manual(values= spf_pals)+
   geom_hline(yintercept = 1)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =18, angle = 45, hjust = 1), 
        axis.text.y = element_text(size=18), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1,
        plot.margin=grid::unit(c(0,0,0,0), "pt"),
        panel.background = element_blank())+
  labs(y = "Normalized Score AUC", x = "")+
  geom_signif(comparisons = list(c("Lacto", "Staph")), color = "black", step = 0.1)

```


## Sepi + Abx data

```{r}

abx_mice = subset(all_spf, Cohort %in% c(29,34))

# because the split is by group. not organism here, we must tweak our standard functions a little

  abx_mice$ID = paste(abx_mice$Group, abx_mice$Cohort, abx_mice$Mouse, sep = "_")
  
  # abx_mice2 = subset(abx_mice, Cohort == 29)
  
  auc_df_dp = abx_mice %>%
    filter(Condition != "Baseline" & Day < 5) %>%
    group_by(Group, ID, Cohort) %>%
    summarise(TEWL_AUC = auc(Day, TEWL, type = "spline"))
  
    auc_df_dp_score = abx_mice %>%
    filter(Day > 1 & Day < 5 & !is.na(complete_score)) %>%
    group_by(Group, ID, Cohort) %>%
    summarise(Score_AUC = auc(Day, complete_score, type = "spline"))

    abx_aucs = left_join(auc_df_dp, auc_df_dp_score)
    
    abx_aucs = subset(abx_aucs, ID %!in% c("ctrl_pbs_29_5", "abx_sepi_29_4", "abx_sepi_29_5")) #this mouse was removed after 1 day and the sepi mice literally died lmao
  
# normalise aucs - to controls

  normalised_abx_auc_df = group_by(filter(abx_aucs, Group == "ctrl_pbs"),Cohort) %>%
    summarise(PBS_TEWLmean = mean(TEWL_AUC, na.rm = TRUE), PBS_scoremean = mean(Score_AUC, na.rm = TRUE)) %>%
    left_join(abx_aucs, ., by = c("Cohort")) %>%
    group_by(Cohort) %>%
    mutate(TEWL_AUC_norm = TEWL_AUC / PBS_TEWLmean,
           Score_AUC_norm = Score_AUC / PBS_scoremean) %>%
    select(Group, ID, Cohort, TEWL_AUC_norm, Score_AUC_norm)
  
  
  # get labels
  
    normalised_abx_auc_df$Group = factor(normalised_abx_auc_df$Group)
    
  df2 = normalised_abx_auc_df
  abx_auc_labs = data.frame()
  PBS_tewl = subset(df2,  Group == "ctrl_pbs")$TEWL_AUC_norm
  PBS_score = subset(df2, Group == "ctrl_pbs")$Score_AUC_norm
  
  for (group in unique(subset(df2, Group != "ctrl_pbs")$Group)){
    org_tewl_norm = subset(df2, Group == group)$TEWL_AUC_norm
    org_score_nrom = subset(df2, Group == group)$Score_AUC_norm
    tewl_p = wilcox.test(PBS_tewl, org_tewl_norm)$p.value
    score_p = wilcox.test(PBS_score, org_score_nrom)$p.value
    abx_auc_labs = rbind(abx_auc_labs, data.frame("Group" = group, "TEWL_p" = tewl_p, "Score_p" = score_p))
  } 


abx_auc_labs = abx_auc_labs %>%
  mutate(tewl_label = case_when(TEWL_p > 0.05 ~ "n.s.",
                                TEWL_p > 0.01 & TEWL_p < 0.05 ~ "*",
                                TEWL_p > 0.001 & TEWL_p < 0.01 ~ "**",
                                TEWL_p > 0 & TEWL_p < 0.001 ~ "***")) %>%
   mutate(score_label = case_when(Score_p > 0.05 ~ "n.s.",
                                Score_p > 0.01 & Score_p < 0.05 ~ "*",
                                Score_p > 0.001 & Score_p < 0.01 ~ "**",
                                Score_p > 0 & Score_p < 0.001 ~ "***")) %>%
   mutate(TEWL_p_sci = format(TEWL_p, scientific = TRUE, digits = 3)) %>%
   mutate(Score_p_sci = format(Score_p, scientific = TRUE, digits = 3))
  

coh29_pals = c("abx_pbs" = "black", "ctrl_pbs" = "grey50", "ctrl_sepi"= "lightblue", "abx_sepi" = "darkblue" )
 



# abx_tewl_auc = 
  ggplot(normalised_abx_auc_df,aes(x = Group, y = TEWL_AUC_norm, color=Group, fill=Group))+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d") + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d")+
  scale_color_manual(values = coh29_pals)+
  scale_fill_manual(values= coh29_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1)+
  # scale_x_discrete(labels = c("PBS", "Antibiotic-treated"))+
  labs(y = "Normalized TEWL AUC", x = "")
  # geom_text(data = abx_auc_labs, mapping = aes(x = Group, y = 1.3, label = round(TEWL_p, 3)), color = "black")



# abx_score_auc = 
  ggplot(normalised_abx_auc_df,aes(x = Group, y = Score_AUC_norm, color=Group, fill=Group))+
  # geom_dotplot(binaxis = "y", stackdir = "center", color = "NA", dotsize = 1.25, alpha  = 0.5)+
  # stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..), width = .3,size = 1.5)+
    stat_summary(fun.y = median, geom = "errorbar", aes(ymax = ..y.., ymin = ..y.., color = "black"),
                 width = .6,size = 1, alpha = 0.9)+
  geom_sina(maxwidth = .7, scale = "area", seed = 1,size = 5, alpha = .5, method = "d", position = position_jitter(width = 0.1, seed = 123)) + 
  geom_sina(maxwidth = .7, scale = "area", seed = 1, size = 5, shape = 1, color = "black", stroke = .8, method = "d", position = position_jitter(width = 0.1, seed = 123))+ # note: points were too close together for sina to handle, had to add jitter
  scale_color_manual(values = coh29_pals)+
  scale_fill_manual(values= coh29_pals)+
  theme_classic()+
  theme(panel.grid.major = element_blank(),
        axis.text.x = element_text(size =12, vjust = 1), 
        axis.text.y = element_text(size=12), axis.title = element_text(size = 20), 
        legend.position = "none",
        aspect.ratio = 1.25/1)+
  labs(y = "Normalized Score AUC", x = "")
  # geom_text(data = abx_auc_labs, mapping = aes(x = Group, y = 2.5, label = round(Score_p, 3)), color = "black")

  


```

## Any bacteria that correlated w severity score beyond s. xylosus in abx mice

```{r}

# There are probably more elegant ways to do this, but alas

corr_analysis_ps = subset_samples(skin_samples, Timepoint == "Endpoint")

# Species-only analysis

corr_analysis_ps_sp = tax_glom(corr_analysis_ps, taxrank= "Species")
corr_analysis_ps_sp = filter_taxa(corr_analysis_ps_sp, function(x) mean(x) > 0.01, TRUE) # taxa greater than 1%

skin_otus = data.frame(t(corr_analysis_ps_sp@otu_table))
taxadf = data.frame(corr_analysis_ps_sp@tax_table)


spf29_auc_norm = subset(normalised_abx_auc_df, Cohort == "29")
spf29_auc_norm$ID  = sapply(spf29_auc_norm$ID, function(x) {paste(strsplit(x, "_")[[1]][c(1,4)], collapse = "_")})
skin_otus$Row.names = rownames(skin_otus)
skin_otus$ID = sapply(skin_otus$Row.names, function(x) {paste(strsplit(x, "_")[[1]][c(3,4)], collapse = "_")})

skin_otus = merge(skin_otus, spf29_auc_norm, by = "ID")

# cor.test(skin_otus[,2], skin_otus$Score, method = "pearson")

cordf_sp = data.frame()
for (col in c(2:4)){
  taxa = colnames(skin_otus)[col]
  taxa = ifelse(startsWith(taxa, "X"), substr(taxa,2,nchar(taxa)), taxa)
  taxa_names = taxadf[rownames(taxadf) == taxa,]
  sciname = paste(taxa_names$Genus, taxa_names$Species, sep = ".")
  cortest = cor.test(skin_otus[,col], skin_otus[,9], method = "pearson")
  cordf_sp = rbind(cordf_sp, data.frame("Taxa" = sciname, "Corr" = as.numeric(cortest$estimate), "P" = cortest$p.value))
}

# Lets do a quick BH correction

cordf_bh = cordf_sp[order(cordf_sp$P),]
cordf_bh$k_m_a = (seq(1,nrow(cordf_bh),1)/nrow(cordf_bh))* 0.05 #this is the critical value
cordf_bh$BH = cordf_bh$P < cordf_bh$k_m_a






```